{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2024-03-26T00:58:30.964337Z",
     "start_time": "2024-03-26T00:58:30.509805Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import copy\n",
    "from ete3 import Tree"
   ]
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting ete3\r\n",
      "  Using cached ete3-3.1.3-py3-none-any.whl\r\n",
      "Installing collected packages: ete3\r\n",
      "Successfully installed ete3-3.1.3\r\n",
      "\r\n",
      "\u001B[1m[\u001B[0m\u001B[34;49mnotice\u001B[0m\u001B[1;39;49m]\u001B[0m\u001B[39;49m A new release of pip is available: \u001B[0m\u001B[31;49m23.2.1\u001B[0m\u001B[39;49m -> \u001B[0m\u001B[32;49m24.0\u001B[0m\r\n",
      "\u001B[1m[\u001B[0m\u001B[34;49mnotice\u001B[0m\u001B[1;39;49m]\u001B[0m\u001B[39;49m To update, run: \u001B[0m\u001B[32;49mpip install --upgrade pip\u001B[0m\r\n"
     ]
    }
   ],
   "source": [
    "!pip install ete3"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T00:58:21.556228Z",
     "start_time": "2024-03-26T00:58:20.093675Z"
    }
   },
   "id": "93129f29e067314c",
   "execution_count": 13
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "file_dir = \"/home/piyumal/PHD/BootstrapSupportPrediction/raw\"\n",
    "mpboot_path = \"/home/piyumal/Tools/MPBOOT/compiled/mpboot\""
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T00:38:58.126962Z",
     "start_time": "2024-03-26T00:38:58.124718Z"
    }
   },
   "id": "d0bbb9696a7a1ecb",
   "execution_count": 4
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########################\n",
      "Running MPBOOT for: 17014_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0\n",
      "Seed:    252323 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:13 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 54 sequences with 2063 columns and 837 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon21  0    (0.000%)\t4\n",
      "2    taxon22  584  (28.308%)\t4\n",
      "3    taxon41  159  (7.707%)\t4\n",
      "4    taxon30  615  (29.811%)\t4\n",
      "5    taxon4   205  (9.937%)\t7\n",
      "6    taxon54  635  (30.780%)\t4\n",
      "7    taxon28  277  (13.427%)\t4\n",
      "8    taxon26  0    (0.000%)\t4\n",
      "9    taxon52  1    (0.048%)\t5\n",
      "10   taxon47  432  (20.940%)\t4\n",
      "11   taxon37  361  (17.499%)\t4\n",
      "12   taxon44  361  (17.499%)\t4\n",
      "13   taxon53  622  (30.150%)\t4\n",
      "14   taxon42  361  (17.499%)\t4\n",
      "15   taxon20  1    (0.048%)\t4\n",
      "16   taxon16  0    (0.000%)\t4\n",
      "17   taxon17  0    (0.000%)\t4\n",
      "18   taxon19  1    (0.048%)\t5\n",
      "19   taxon18  0    (0.000%)\t4\n",
      "20   taxon40  0    (0.000%)\t4\n",
      "21   taxon48  158  (7.659%)\t10\n",
      "22   taxon45  630  (30.538%)\t4\n",
      "23   taxon35  3    (0.145%)\t6\n",
      "24   taxon23  0    (0.000%)\t4\n",
      "25   taxon38  0    (0.000%)\t4\n",
      "26   taxon51  615  (29.811%)\t5\n",
      "27   taxon15  655  (31.750%)\t5\n",
      "28   taxon29  0    (0.000%)\t4\n",
      "29   taxon32  584  (28.308%)\t4\n",
      "30   taxon33  0    (0.000%)\t4\n",
      "31   taxon7   584  (28.308%)\t4\n",
      "32   taxon1   0    (0.000%)\t4\n",
      "33   taxon50  421  (20.407%)\t4\n",
      "34   taxon6   0    (0.000%)\t4\n",
      "35   taxon5   657  (31.847%)\t4\n",
      "36   taxon24  0    (0.000%)\t4\n",
      "37   taxon8   621  (30.102%)\t5\n",
      "38   taxon36  1    (0.048%)\t5\n",
      "39   taxon2   584  (28.308%)\t4\n",
      "40   taxon3   621  (30.102%)\t4\n",
      "41   taxon39  361  (17.499%)\t4\n",
      "42   taxon27  424  (20.553%)\t6\n",
      "43   taxon14  1    (0.048%)\t5\n",
      "44   taxon13  1    (0.048%)\t5\n",
      "45   taxon10  1    (0.048%)\t5\n",
      "46   taxon11  614  (29.762%)\t4\n",
      "47   taxon9   0    (0.000%)\t4\n",
      "48   taxon12  615  (29.811%)\t4\n",
      "49   taxon43  365  (17.693%)\t5\n",
      "50   taxon34  614  (29.762%)\t4\n",
      "51   taxon49  643  (31.168%)\t4\n",
      "52   taxon31  275  (13.330%)\t4\n",
      "53   taxon46  389  (18.856%)\t7\n",
      "54   taxon25  1    (0.048%)\t4\n",
      "**** TOTAL    15053 (13.512%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 4695\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 7.793 MB RAM is required!\n",
      "Current best tree score: 4695\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "NOTE: REPS computation is segmented into 2 parts for speed...\n",
      "segment#1 ends at 432\n",
      "segment#2 ends at 740\n",
      "0 seconds.\n",
      "Current best score: 4647 / CPU time: 1\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 1: 4638 / CPU time: 1s\n",
      "RATCHET Iteration 10 / Score: 4639 / Time: 0h:0m:1s\n",
      "BETTER TREE FOUND at iteration 17: 4636 / CPU time: 2s\n",
      "\n",
      "RATCHET Iteration 20 / Score: 4638 / Time: 0h:0m:1s (0h:0m:8s left)\n",
      "RATCHET Iteration 30 / Score: 4636 / Time: 0h:0m:1s (0h:0m:5s left)\n",
      "RATCHET Iteration 40 / Score: 4632 / Time: 0h:0m:2s (0h:0m:4s left)\n",
      "BETTER TREE FOUND at iteration 40: 4632 / CPU time: 2s\n",
      "RATCHET Iteration 50 / Score: 4636 / Time: 0h:0m:2s (0h:0m:4s left)\n",
      "RATCHET Iteration 60 / Score: 4632 / Time: 0h:0m:2s (0h:0m:3s left)\n",
      "RATCHET Iteration 70 / Score: 4634 / Time: 0h:0m:3s (0h:0m:3s left)\n",
      "RATCHET Iteration 80 / Score: 4632 / Time: 0h:0m:3s (0h:0m:2s left)\n",
      "RATCHET Iteration 90 / Score: 4632 / Time: 0h:0m:3s (0h:0m:2s left)\n",
      "RATCHET Iteration 100 / Score: 4632 / Time: 0h:0m:4s (0h:0m:1s left)\n",
      "RATCHET Iteration 110 / Score: 4631 / Time: 0h:0m:4s (0h:0m:1s left)\n",
      "BETTER TREE FOUND at iteration 110: 4631 / CPU time: 4s\n",
      "RATCHET Iteration 120 / Score: 4632 / Time: 0h:0m:4s (0h:0m:3s left)\n",
      "RATCHET Iteration 130 / Score: 4632 / Time: 0h:0m:4s (0h:0m:3s left)\n",
      "RATCHET Iteration 140 / Score: 4632 / Time: 0h:0m:5s (0h:0m:2s left)\n",
      "RATCHET Iteration 150 / Score: 4631 / Time: 0h:0m:5s (0h:0m:2s left)\n",
      "RATCHET Iteration 160 / Score: 4631 / Time: 0h:0m:5s (0h:0m:1s left)\n",
      "RATCHET Iteration 170 / Score: 4631 / Time: 0h:0m:5s (0h:0m:1s left)\n",
      "RATCHET Iteration 180 / Score: 4631 / Time: 0h:0m:6s (0h:0m:1s left)\n",
      "RATCHET Iteration 190 / Score: 4634 / Time: 0h:0m:6s (0h:0m:0s left)\n",
      "RATCHET Iteration 200 / Score: 4631 / Time: 0h:0m:6s (0h:0m:0s left)\n",
      "RATCHET Iteration 210 / Score: 4631 / Time: 0h:0m:6s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  5 sec.\n",
      "NOTE: At the end, 43331 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "4631 4631 4631 4631 4632 \n",
      "BEST SCORE FOUND : 4631\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.splits.nex\n",
      "\n",
      "CPU time used for tree search: 11.764 sec (0h:0m:11s)\n",
      "Wall-clock time used for tree search: 11.770 sec (0h:0m:11s)\n",
      "Total CPU time used: 11.828 sec (0h:0m:11s)\n",
      "Total wall-clock time used: 11.846 sec (0h:0m:11s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.splits.nex...\n",
      "54 taxa and 4531 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.contree\n",
      "Parsimony score of consensus tree: 4648\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/17014_0/17014_0.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:34:25 2024\n",
      "Completed MPBOOT for: 17014_0\n",
      "##########################\n",
      "Running MPBOOT for: 14141_2\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/14141_2/14141_2_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/14141_2/14141_2.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/14141_2/14141_2\n",
      "Seed:    131818 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:25 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14141_2/14141_2_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "WARNING: 3 sites contain only gaps or ambiguous chars.\n",
      "Alignment has 33 sequences with 377 columns and 244 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon9   43   (11.406%)\t4\n",
      "2    taxon28  72   (19.098%)\t4\n",
      "3    taxon35  41   (10.875%)\t4\n",
      "4    taxon73  33   (8.753%)\t4\n",
      "5    taxon3   33   (8.753%)\t4\n",
      "6    taxon61  33   (8.753%)\t4\n",
      "7    taxon65  34   (9.019%)\t4\n",
      "8    taxon10  53   (14.058%)\t4\n",
      "9    taxon11  71   (18.833%)\t4\n",
      "10   taxon27  49   (12.997%)\t4\n",
      "11   taxon41  46   (12.202%)\t4\n",
      "12   taxon39  46   (12.202%)\t4\n",
      "13   taxon58  64   (16.976%)\t4\n",
      "14   taxon48  64   (16.976%)\t4\n",
      "15   taxon43  85   (22.546%)\t4\n",
      "16   taxon46  88   (23.342%)\t4\n",
      "17   taxon36  83   (22.016%)\t4\n",
      "18   taxon37  83   (22.016%)\t4\n",
      "19   taxon38  83   (22.016%)\t4\n",
      "20   taxon33  66   (17.507%)\t4\n",
      "21   taxon26  46   (12.202%)\t4\n",
      "22   taxon14  48   (12.732%)\t4\n",
      "23   taxon18  67   (17.772%)\t4\n",
      "24   taxon16  47   (12.467%)\t4\n",
      "25   taxon23  47   (12.467%)\t4\n",
      "26   taxon15  96   (25.464%)\t4\n",
      "27   taxon17  97   (25.729%)\t4\n",
      "28   taxon32  95   (25.199%)\t4\n",
      "29   taxon7   58   (15.385%)\t4\n",
      "30   taxon4   45   (11.936%)\t4\n",
      "31   taxon6   45   (11.936%)\t4\n",
      "32   taxon8   45   (11.936%)\t4\n",
      "33   taxon64  33   (8.753%)\t4\n",
      "**** TOTAL    1939 (15.586%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 439\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14141_2/14141_2.newick ...\n",
      "NOTE: Changing to old kernel as input tree is multifurcating\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 1.508 MB RAM is required!\n",
      "ERROR: A taxon has no name. (line 1 column 1365)\n",
      "Completed MPBOOT for: 14141_2\n",
      "##########################\n",
      "Running MPBOOT for: 20937_2\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2\n",
      "Seed:    136082 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:25 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 112 sequences with 1901 columns and 603 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon139  83   (4.366%)\t4\n",
      "2    taxon146  83   (4.366%)\t4\n",
      "3    taxon145  83   (4.366%)\t4\n",
      "4    taxon143  86   (4.524%)\t4\n",
      "5    taxon25   82   (4.314%)\t4\n",
      "6    taxon88   80   (4.208%)\t4\n",
      "7    taxon87   80   (4.208%)\t4\n",
      "8    taxon89   120  (6.312%)\t4\n",
      "9    taxon85   80   (4.208%)\t4\n",
      "10   taxon122  58   (3.051%)\t4\n",
      "11   taxon110  92   (4.840%)\t4\n",
      "12   taxon121  78   (4.103%)\t4\n",
      "13   taxon155  79   (4.156%)\t4\n",
      "14   taxon154  80   (4.208%)\t4\n",
      "15   taxon153  83   (4.366%)\t4\n",
      "16   taxon156  80   (4.208%)\t4\n",
      "17   taxon24   79   (4.156%)\t4\n",
      "18   taxon11   78   (4.103%)\t4\n",
      "19   taxon21   78   (4.103%)\t4\n",
      "20   taxon23   78   (4.103%)\t4\n",
      "21   taxon18   78   (4.103%)\t4\n",
      "22   taxon15   79   (4.156%)\t4\n",
      "23   taxon22   77   (4.050%)\t4\n",
      "24   taxon35   77   (4.050%)\t4\n",
      "25   taxon95   81   (4.261%)\t4\n",
      "26   taxon99   78   (4.103%)\t4\n",
      "27   taxon100  81   (4.261%)\t4\n",
      "28   taxon94   80   (4.208%)\t4\n",
      "29   taxon98   79   (4.156%)\t4\n",
      "30   taxon96   85   (4.471%)\t4\n",
      "31   taxon97   79   (4.156%)\t4\n",
      "32   taxon101  74   (3.893%)\t4\n",
      "33   taxon93   103  (5.418%)\t4\n",
      "34   taxon3    81   (4.261%)\t4\n",
      "35   taxon2    80   (4.208%)\t4\n",
      "36   taxon157  80   (4.208%)\t4\n",
      "37   taxon10   81   (4.261%)\t4\n",
      "38   taxon1    162  (8.522%)\t4\n",
      "39   taxon9    81   (4.261%)\t4\n",
      "40   taxon4    82   (4.314%)\t4\n",
      "41   taxon54   82   (4.314%)\t4\n",
      "42   taxon36   78   (4.103%)\t4\n",
      "43   taxon37   77   (4.050%)\t4\n",
      "44   taxon126  95   (4.997%)\t4\n",
      "45   taxon117  202  (10.626%)\t4\n",
      "46   taxon116  93   (4.892%)\t4\n",
      "47   taxon137  81   (4.261%)\t4\n",
      "48   taxon59   82   (4.314%)\t4\n",
      "49   taxon55   82   (4.314%)\t4\n",
      "50   taxon76   81   (4.261%)\t4\n",
      "51   taxon81   83   (4.366%)\t4\n",
      "52   taxon63   82   (4.314%)\t4\n",
      "53   taxon70   81   (4.261%)\t4\n",
      "54   taxon66   82   (4.314%)\t4\n",
      "55   taxon64   82   (4.314%)\t4\n",
      "56   taxon68   81   (4.261%)\t4\n",
      "57   taxon77   81   (4.261%)\t4\n",
      "58   taxon60   82   (4.314%)\t4\n",
      "59   taxon71   81   (4.261%)\t4\n",
      "60   taxon82   83   (4.366%)\t4\n",
      "61   taxon84   82   (4.314%)\t4\n",
      "62   taxon83   82   (4.314%)\t4\n",
      "63   taxon78   83   (4.366%)\t4\n",
      "64   taxon62   82   (4.314%)\t4\n",
      "65   taxon80   83   (4.366%)\t4\n",
      "66   taxon65   82   (4.314%)\t4\n",
      "67   taxon79   83   (4.366%)\t4\n",
      "68   taxon67   81   (4.261%)\t4\n",
      "69   taxon73   81   (4.261%)\t4\n",
      "70   taxon119  68   (3.577%)\t4\n",
      "71   taxon34   79   (4.156%)\t4\n",
      "72   taxon124  96   (5.050%)\t4\n",
      "73   taxon125  94   (4.945%)\t4\n",
      "74   taxon53   81   (4.261%)\t4\n",
      "75   taxon49   81   (4.261%)\t4\n",
      "76   taxon46   78   (4.103%)\t4\n",
      "77   taxon39   81   (4.261%)\t4\n",
      "78   taxon47   81   (4.261%)\t4\n",
      "79   taxon50   78   (4.103%)\t4\n",
      "80   taxon44   81   (4.261%)\t4\n",
      "81   taxon41   81   (4.261%)\t4\n",
      "82   taxon45   81   (4.261%)\t4\n",
      "83   taxon115  91   (4.787%)\t4\n",
      "84   taxon90   81   (4.261%)\t4\n",
      "85   taxon91   81   (4.261%)\t4\n",
      "86   taxon26   87   (4.577%)\t4\n",
      "87   taxon28   86   (4.524%)\t4\n",
      "88   taxon27   85   (4.471%)\t4\n",
      "89   taxon29   78   (4.103%)\t4\n",
      "90   taxon130  86   (4.524%)\t4\n",
      "91   taxon135  86   (4.524%)\t4\n",
      "92   taxon128  86   (4.524%)\t4\n",
      "93   taxon131  86   (4.524%)\t4\n",
      "94   taxon147  133  (6.996%)\t4\n",
      "95   taxon149  86   (4.524%)\t4\n",
      "96   taxon152  83   (4.366%)\t4\n",
      "97   taxon148  86   (4.524%)\t4\n",
      "98   taxon150  83   (4.366%)\t4\n",
      "99   taxon151  83   (4.366%)\t4\n",
      "100  taxon92   80   (4.208%)\t4\n",
      "101  taxon109  78   (4.103%)\t4\n",
      "102  taxon106  79   (4.156%)\t4\n",
      "103  taxon105  78   (4.103%)\t4\n",
      "104  taxon104  79   (4.156%)\t5\n",
      "105  taxon102  79   (4.156%)\t4\n",
      "106  taxon30   96   (5.050%)\t4\n",
      "107  taxon31   78   (4.103%)\t4\n",
      "108  taxon118  94   (4.945%)\t4\n",
      "109  taxon123  129  (6.786%)\t4\n",
      "110  taxon113  85   (4.471%)\t4\n",
      "111  taxon112  83   (4.366%)\t4\n",
      "112  taxon114  83   (4.366%)\t4\n",
      "**** TOTAL     9516 (4.469%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 1738\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 11.820 MB RAM is required!\n",
      "Current best tree score: 1738\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 1730 / CPU time: 1\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 1: 1729 / CPU time: 1s\n",
      "RATCHET Iteration 10 / Score: 1729 / Time: 0h:0m:1s\n",
      "RATCHET Iteration 20 / Score: 1729 / Time: 0h:0m:2s (0h:0m:20s left)\n",
      "RATCHET Iteration 30 / Score: 1729 / Time: 0h:0m:2s (0h:0m:16s left)\n",
      "RATCHET Iteration 40 / Score: 1729 / Time: 0h:0m:3s (0h:0m:13s left)\n",
      "RATCHET Iteration 50 / Score: 1729 / Time: 0h:0m:3s (0h:0m:12s left)\n",
      "RATCHET Iteration 60 / Score: 1729 / Time: 0h:0m:4s (0h:0m:10s left)\n",
      "RATCHET Iteration 70 / Score: 1729 / Time: 0h:0m:5s (0h:0m:9s left)\n",
      "RATCHET Iteration 80 / Score: 1729 / Time: 0h:0m:5s (0h:0m:8s left)\n",
      "RATCHET Iteration 90 / Score: 1729 / Time: 0h:0m:6s (0h:0m:7s left)\n",
      "RATCHET Iteration 100 / Score: 1729 / Time: 0h:0m:6s (0h:0m:6s left)\n",
      "RATCHET Iteration 110 / Score: 1729 / Time: 0h:0m:7s (0h:0m:6s left)\n",
      "RATCHET Iteration 120 / Score: 1729 / Time: 0h:0m:7s (0h:0m:5s left)\n",
      "RATCHET Iteration 130 / Score: 1729 / Time: 0h:0m:8s (0h:0m:4s left)\n",
      "RATCHET Iteration 140 / Score: 1729 / Time: 0h:0m:8s (0h:0m:3s left)\n",
      "RATCHET Iteration 150 / Score: 1729 / Time: 0h:0m:9s (0h:0m:3s left)\n",
      "RATCHET Iteration 160 / Score: 1729 / Time: 0h:0m:10s (0h:0m:2s left)\n",
      "RATCHET Iteration 170 / Score: 1729 / Time: 0h:0m:10s (0h:0m:1s left)\n",
      "RATCHET Iteration 180 / Score: 1729 / Time: 0h:0m:11s (0h:0m:1s left)\n",
      "RATCHET Iteration 190 / Score: 1729 / Time: 0h:0m:11s (0h:0m:0s left)\n",
      "RATCHET Iteration 200 / Score: 1729 / Time: 0h:0m:12s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  8 sec.\n",
      "NOTE: At the end, 93980 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "1729 1729 1729 1729 1729 \n",
      "BEST SCORE FOUND : 1729\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.splits.nex\n",
      "\n",
      "CPU time used for tree search: 19.934 sec (0h:0m:19s)\n",
      "Wall-clock time used for tree search: 19.940 sec (0h:0m:19s)\n",
      "Total CPU time used: 20.043 sec (0h:0m:20s)\n",
      "Total wall-clock time used: 20.066 sec (0h:0m:20s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.splits.nex...\n",
      "112 taxa and 1203 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.contree\n",
      "Parsimony score of consensus tree: 1729\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/20937_2/20937_2.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:34:45 2024\n",
      "Completed MPBOOT for: 20937_2\n",
      "##########################\n",
      "Running MPBOOT for: 12687_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0\n",
      "Seed:    237852 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:45 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 60 sequences with 494 columns and 342 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon39  23   (4.656%)\t4\n",
      "2    taxon41  24   (4.858%)\t4\n",
      "3    taxon40  23   (4.656%)\t4\n",
      "4    taxon43  22   (4.453%)\t4\n",
      "5    taxon45  45   (9.109%)\t4\n",
      "6    taxon44  43   (8.704%)\t4\n",
      "7    taxon19  9    (1.822%)\t4\n",
      "8    taxon18  11   (2.227%)\t4\n",
      "9    taxon32  44   (8.907%)\t4\n",
      "10   taxon37  17   (3.441%)\t4\n",
      "11   taxon38  15   (3.036%)\t4\n",
      "12   taxon2   1    (0.202%)\t4\n",
      "13   taxon67  3    (0.607%)\t4\n",
      "14   taxon1   0    (0.000%)\t4\n",
      "15   taxon9   2    (0.405%)\t4\n",
      "16   taxon62  2    (0.405%)\t4\n",
      "17   taxon6   4    (0.810%)\t4\n",
      "18   taxon4   4    (0.810%)\t4\n",
      "19   taxon5   4    (0.810%)\t4\n",
      "20   taxon3   15   (3.036%)\t4\n",
      "21   taxon63  3    (0.607%)\t4\n",
      "22   taxon65  4    (0.810%)\t4\n",
      "23   taxon66  4    (0.810%)\t4\n",
      "24   taxon7   2    (0.405%)\t4\n",
      "25   taxon31  44   (8.907%)\t4\n",
      "26   taxon58  18   (3.644%)\t4\n",
      "27   taxon56  18   (3.644%)\t4\n",
      "28   taxon59  18   (3.644%)\t4\n",
      "29   taxon46  33   (6.680%)\t4\n",
      "30   taxon47  34   (6.883%)\t4\n",
      "31   taxon50  32   (6.478%)\t4\n",
      "32   taxon51  32   (6.478%)\t4\n",
      "33   taxon23  7    (1.417%)\t4\n",
      "34   taxon22  6    (1.215%)\t4\n",
      "35   taxon11  13   (2.632%)\t4\n",
      "36   taxon13  14   (2.834%)\t4\n",
      "37   taxon12  14   (2.834%)\t4\n",
      "38   taxon14  13   (2.632%)\t4\n",
      "39   taxon15  14   (2.834%)\t4\n",
      "40   taxon16  28   (5.668%)\t4\n",
      "41   taxon17  12   (2.429%)\t4\n",
      "42   taxon26  18   (3.644%)\t4\n",
      "43   taxon25  6    (1.215%)\t4\n",
      "44   taxon27  10   (2.024%)\t4\n",
      "45   taxon28  7    (1.417%)\t4\n",
      "46   taxon24  10   (2.024%)\t4\n",
      "47   taxon61  18   (3.644%)\t4\n",
      "48   taxon60  17   (3.441%)\t4\n",
      "49   taxon29  34   (6.883%)\t4\n",
      "50   taxon30  32   (6.478%)\t4\n",
      "51   taxon21  19   (3.846%)\t4\n",
      "52   taxon20  20   (4.049%)\t4\n",
      "53   taxon55  45   (9.109%)\t4\n",
      "54   taxon54  39   (7.895%)\t4\n",
      "55   taxon49  37   (7.490%)\t4\n",
      "56   taxon48  35   (7.085%)\t4\n",
      "57   taxon53  34   (6.883%)\t4\n",
      "58   taxon52  34   (6.883%)\t4\n",
      "59   taxon36  29   (5.870%)\t4\n",
      "60   taxon35  12   (2.429%)\t4\n",
      "**** TOTAL    1130 (3.812%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 1509\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 3.576 MB RAM is required!\n",
      "Current best tree score: 1509\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 1497 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 1: 1496 / CPU time: 0s\n",
      "\n",
      "BETTER TREE FOUND at iteration 3: 1495 / CPU time: 0s\n",
      "RATCHET Iteration 10 / Score: 1495 / Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 18: 1494 / CPU time: 1s\n",
      "RATCHET Iteration 20 / Score: 1495 / Time: 0h:0m:0s (0h:0m:4s left)\n",
      "RATCHET Iteration 30 / Score: 1494 / Time: 0h:0m:1s (0h:0m:3s left)\n",
      "RATCHET Iteration 40 / Score: 1494 / Time: 0h:0m:1s (0h:0m:2s left)\n",
      "RATCHET Iteration 50 / Score: 1494 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 60 / Score: 1494 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 70 / Score: 1494 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 80 / Score: 1494 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 1494 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 1494 / Time: 0h:0m:2s (0h:0m:0s left)\n",
      "RATCHET Iteration 110 / Score: 1494 / Time: 0h:0m:2s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  2 sec.\n",
      "NOTE: At the end, 30792 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "1494 1494 1494 1494 1494 \n",
      "BEST SCORE FOUND : 1494\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.splits.nex\n",
      "\n",
      "CPU time used for tree search: 4.491 sec (0h:0m:4s)\n",
      "Wall-clock time used for tree search: 4.496 sec (0h:0m:4s)\n",
      "Total CPU time used: 4.538 sec (0h:0m:4s)\n",
      "Total wall-clock time used: 4.551 sec (0h:0m:4s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.splits.nex...\n",
      "60 taxa and 405 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.contree\n",
      "Parsimony score of consensus tree: 1498\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/12687_0/12687_0.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:34:49 2024\n",
      "Completed MPBOOT for: 12687_0\n",
      "##########################\n",
      "Running MPBOOT for: 14725_7\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7\n",
      "Seed:    806407 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:49 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 40 sequences with 844 columns and 583 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon2   246  (29.147%)\t5\n",
      "2    taxon10  178  (21.090%)\t4\n",
      "3    taxon26  176  (20.853%)\t4\n",
      "4    taxon4   166  (19.668%)\t4\n",
      "5    taxon17  92   (10.900%)\t4\n",
      "6    taxon16  157  (18.602%)\t7\n",
      "7    taxon24  99   (11.730%)\t4\n",
      "8    taxon25  258  (30.569%)\t4\n",
      "9    taxon9   248  (29.384%)\t4\n",
      "10   taxon5   243  (28.791%)\t4\n",
      "11   taxon28  103  (12.204%)\t4\n",
      "12   taxon3   210  (24.882%)\t4\n",
      "13   taxon29  229  (27.133%)\t4\n",
      "14   taxon30  458  (54.265%) !!!\t4\n",
      "15   taxon27  260  (30.806%)\t4\n",
      "16   taxon7   217  (25.711%)\t4\n",
      "17   taxon36  96   (11.374%)\t4\n",
      "18   taxon37  97   (11.493%)\t5\n",
      "19   taxon38  111  (13.152%)\t4\n",
      "20   taxon35  95   (11.256%)\t4\n",
      "21   taxon1   235  (27.844%)\t4\n",
      "22   taxon31  104  (12.322%)\t4\n",
      "23   taxon40  129  (15.284%)\t4\n",
      "24   taxon23  100  (11.848%)\t4\n",
      "25   taxon22  253  (29.976%)\t4\n",
      "26   taxon19  100  (11.848%)\t4\n",
      "27   taxon21  100  (11.848%)\t4\n",
      "28   taxon20  100  (11.848%)\t4\n",
      "29   taxon34  174  (20.616%)\t5\n",
      "30   taxon6   283  (33.531%)\t4\n",
      "31   taxon14  201  (23.815%)\t4\n",
      "32   taxon33  185  (21.919%)\t4\n",
      "33   taxon13  214  (25.355%)\t4\n",
      "34   taxon32  160  (18.957%)\t4\n",
      "35   taxon11  176  (20.853%)\t4\n",
      "36   taxon12  216  (25.592%)\t4\n",
      "37   taxon15  149  (17.654%)\t4\n",
      "38   taxon8   193  (22.867%)\t4\n",
      "39   taxon39  163  (19.313%)\t4\n",
      "40   taxon18  251  (29.739%)\t4\n",
      "WARNING: 1 sequences contain more than 50% gaps/ambiguity\n",
      "**** TOTAL    7225 (21.401%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 1717\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 3.986 MB RAM is required!\n",
      "Current best tree score: 1717\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 1698 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 1: 1697 / CPU time: 0s\n",
      "RATCHET Iteration 10 / Score: 1699 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 1698 / Time: 0h:0m:0s (0h:0m:2s left)\n",
      "RATCHET Iteration 30 / Score: 1697 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "RATCHET Iteration 40 / Score: 1699 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "RATCHET Iteration 50 / Score: 1697 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 1697 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 1698 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 1697 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 1698 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 1698 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  2 sec.\n",
      "NOTE: At the end, 14357 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "1697 1698 1698 1698 1698 \n",
      "BEST SCORE FOUND : 1697\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.splits.nex\n",
      "\n",
      "CPU time used for tree search: 3.380 sec (0h:0m:3s)\n",
      "Wall-clock time used for tree search: 3.385 sec (0h:0m:3s)\n",
      "Total CPU time used: 3.415 sec (0h:0m:3s)\n",
      "Total wall-clock time used: 3.424 sec (0h:0m:3s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.splits.nex...\n",
      "40 taxa and 760 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.contree\n",
      "Parsimony score of consensus tree: 1700\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/14725_7/14725_7.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:34:53 2024\n",
      "Completed MPBOOT for: 14725_7\n",
      "##########################\n",
      "Running MPBOOT for: 12162_5\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5\n",
      "Seed:    245155 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:53 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "WARNING: 3 sites contain only gaps or ambiguous chars.\n",
      "Alignment has 67 sequences with 790 columns and 218 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon43  25   (3.165%)\t4\n",
      "2    taxon22  14   (1.772%)\t4\n",
      "3    taxon29  14   (1.772%)\t4\n",
      "4    taxon26  14   (1.772%)\t4\n",
      "5    taxon30  14   (1.772%)\t4\n",
      "6    taxon38  14   (1.772%)\t4\n",
      "7    taxon27  14   (1.772%)\t4\n",
      "8    taxon24  14   (1.772%)\t4\n",
      "9    taxon25  14   (1.772%)\t4\n",
      "10   taxon23  14   (1.772%)\t4\n",
      "11   taxon28  14   (1.772%)\t4\n",
      "12   taxon21  14   (1.772%)\t4\n",
      "13   taxon34  14   (1.772%)\t4\n",
      "14   taxon31  14   (1.772%)\t4\n",
      "15   taxon32  14   (1.772%)\t4\n",
      "16   taxon33  14   (1.772%)\t4\n",
      "17   taxon35  14   (1.772%)\t4\n",
      "18   taxon37  14   (1.772%)\t4\n",
      "19   taxon36  15   (1.899%)\t4\n",
      "20   taxon39  15   (1.899%)\t4\n",
      "21   taxon40  64   (8.101%)\t4\n",
      "22   taxon45  23   (2.911%)\t4\n",
      "23   taxon46  45   (5.696%)\t4\n",
      "24   taxon51  14   (1.772%)\t4\n",
      "25   taxon18  14   (1.772%)\t4\n",
      "26   taxon15  14   (1.772%)\t4\n",
      "27   taxon14  13   (1.646%)\t4\n",
      "28   taxon16  12   (1.519%)\t4\n",
      "29   taxon17  13   (1.646%)\t4\n",
      "30   taxon41  26   (3.291%)\t4\n",
      "31   taxon42  24   (3.038%)\t4\n",
      "32   taxon44  25   (3.165%)\t4\n",
      "33   taxon5   14   (1.772%)\t4\n",
      "34   taxon7   14   (1.772%)\t4\n",
      "35   taxon8   14   (1.772%)\t4\n",
      "36   taxon6   14   (1.772%)\t4\n",
      "37   taxon12  14   (1.772%)\t4\n",
      "38   taxon9   14   (1.772%)\t4\n",
      "39   taxon10  14   (1.772%)\t4\n",
      "40   taxon68  13   (1.646%)\t4\n",
      "41   taxon63  14   (1.772%)\t4\n",
      "42   taxon65  14   (1.772%)\t4\n",
      "43   taxon64  14   (1.772%)\t4\n",
      "44   taxon2   14   (1.772%)\t4\n",
      "45   taxon1   14   (1.772%)\t4\n",
      "46   taxon4   14   (1.772%)\t4\n",
      "47   taxon3   14   (1.772%)\t4\n",
      "48   taxon11  14   (1.772%)\t4\n",
      "49   taxon66  14   (1.772%)\t4\n",
      "50   taxon67  14   (1.772%)\t4\n",
      "51   taxon62  15   (1.899%)\t4\n",
      "52   taxon61  15   (1.899%)\t4\n",
      "53   taxon60  15   (1.899%)\t4\n",
      "54   taxon59  25   (3.165%)\t4\n",
      "55   taxon52  15   (1.899%)\t4\n",
      "56   taxon13  14   (1.772%)\t4\n",
      "57   taxon20  25   (3.165%)\t4\n",
      "58   taxon19  14   (1.772%)\t4\n",
      "59   taxon50  14   (1.772%)\t4\n",
      "60   taxon53  15   (1.899%)\t4\n",
      "61   taxon58  15   (1.899%)\t4\n",
      "62   taxon54  15   (1.899%)\t4\n",
      "63   taxon56  15   (1.899%)\t4\n",
      "64   taxon55  15   (1.899%)\t4\n",
      "65   taxon47  26   (3.291%)\t4\n",
      "66   taxon48  62   (7.848%)\t4\n",
      "67   taxon49  16   (2.025%)\t4\n",
      "**** TOTAL    1162 (2.195%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 459\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 2.573 MB RAM is required!\n",
      "Current best tree score: 459\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 454 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 2: 453 / CPU time: 0s\n",
      "RATCHET Iteration 10 / Score: 453 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 453 / Time: 0h:0m:0s (0h:0m:3s left)\n",
      "RATCHET Iteration 30 / Score: 453 / Time: 0h:0m:1s (0h:0m:2s left)\n",
      "RATCHET Iteration 40 / Score: 453 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 50 / Score: 453 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 60 / Score: 453 / Time: 0h:0m:1s (0h:0m:1s left)\n",
      "RATCHET Iteration 70 / Score: 453 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 453 / Time: 0h:0m:2s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 453 / Time: 0h:0m:2s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 453 / Time: 0h:0m:2s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  3 sec.\n",
      "NOTE: At the end, 79684 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "453 453 453 453 453 \n",
      "BEST SCORE FOUND : 453\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.splits.nex\n",
      "\n",
      "CPU time used for tree search: 5.035 sec (0h:0m:5s)\n",
      "Wall-clock time used for tree search: 5.036 sec (0h:0m:5s)\n",
      "Total CPU time used: 5.094 sec (0h:0m:5s)\n",
      "Total wall-clock time used: 5.103 sec (0h:0m:5s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.splits.nex...\n",
      "67 taxa and 3455 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.contree\n",
      "Parsimony score of consensus tree: 453\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/12162_5/12162_5.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:34:58 2024\n",
      "Completed MPBOOT for: 12162_5\n",
      "##########################\n",
      "Running MPBOOT for: 15861_1\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/15861_1/15861_1_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/15861_1/15861_1.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/15861_1/15861_1\n",
      "Seed:    376230 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:34:58 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15861_1/15861_1_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains protein sequences\n",
      "Alignment has 433 sequences with 90 columns and 89 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon403  20   (22.222%)\t19\n",
      "2    taxon279  20   (22.222%)\t19\n",
      "3    taxon117  20   (22.222%)\t20\n",
      "4    taxon118  24   (26.667%)\t20\n",
      "5    taxon316  20   (22.222%)\t19\n",
      "6    taxon260  20   (22.222%)\t20\n",
      "7    taxon264  20   (22.222%)\t19\n",
      "8    taxon132  20   (22.222%)\t18\n",
      "9    taxon429  20   (22.222%)\t19\n",
      "10   taxon164  20   (22.222%)\t18\n",
      "11   taxon25   20   (22.222%)\t19\n",
      "12   taxon19   22   (24.444%)\t19\n",
      "13   taxon227  20   (22.222%)\t20\n",
      "14   taxon418  20   (22.222%)\t20\n",
      "15   taxon405  20   (22.222%)\t19\n",
      "16   taxon329  20   (22.222%)\t19\n",
      "17   taxon221  20   (22.222%)\t18\n",
      "18   taxon306  20   (22.222%)\t19\n",
      "19   taxon68   20   (22.222%)\t19\n",
      "20   taxon67   20   (22.222%)\t19\n",
      "21   taxon364  20   (22.222%)\t18\n",
      "22   taxon183  20   (22.222%)\t20\n",
      "23   taxon420  20   (22.222%)\t18\n",
      "24   taxon150  20   (22.222%)\t18\n",
      "25   taxon436  20   (22.222%)\t17\n",
      "26   taxon369  20   (22.222%)\t20\n",
      "27   taxon299  20   (22.222%)\t19\n",
      "28   taxon334  20   (22.222%)\t19\n",
      "29   taxon370  20   (22.222%)\t20\n",
      "30   taxon190  20   (22.222%)\t18\n",
      "31   taxon34   20   (22.222%)\t19\n",
      "32   taxon233  20   (22.222%)\t20\n",
      "33   taxon165  20   (22.222%)\t18\n",
      "34   taxon24   20   (22.222%)\t19\n",
      "35   taxon283  20   (22.222%)\t19\n",
      "36   taxon124  20   (22.222%)\t19\n",
      "37   taxon428  20   (22.222%)\t20\n",
      "38   taxon135  20   (22.222%)\t20\n",
      "39   taxon356  20   (22.222%)\t20\n",
      "40   taxon253  20   (22.222%)\t19\n",
      "41   taxon258  20   (22.222%)\t20\n",
      "42   taxon399  20   (22.222%)\t20\n",
      "43   taxon348  20   (22.222%)\t19\n",
      "44   taxon372  20   (22.222%)\t19\n",
      "45   taxon357  20   (22.222%)\t19\n",
      "46   taxon281  20   (22.222%)\t18\n",
      "47   taxon179  20   (22.222%)\t19\n",
      "48   taxon157  20   (22.222%)\t18\n",
      "49   taxon27   20   (22.222%)\t19\n",
      "50   taxon126  20   (22.222%)\t18\n",
      "51   taxon425  20   (22.222%)\t19\n",
      "52   taxon333  20   (22.222%)\t19\n",
      "53   taxon392  20   (22.222%)\t20\n",
      "54   taxon206  20   (22.222%)\t20\n",
      "55   taxon238  20   (22.222%)\t18\n",
      "56   taxon145  20   (22.222%)\t19\n",
      "57   taxon23   20   (22.222%)\t19\n",
      "58   taxon146  20   (22.222%)\t19\n",
      "59   taxon6    20   (22.222%)\t19\n",
      "60   taxon371  20   (22.222%)\t19\n",
      "61   taxon210  20   (22.222%)\t20\n",
      "62   taxon373  20   (22.222%)\t19\n",
      "63   taxon177  20   (22.222%)\t17\n",
      "64   taxon35   20   (22.222%)\t19\n",
      "65   taxon54   20   (22.222%)\t20\n",
      "66   taxon305  20   (22.222%)\t19\n",
      "67   taxon192  20   (22.222%)\t18\n",
      "68   taxon15   20   (22.222%)\t18\n",
      "69   taxon151  20   (22.222%)\t19\n",
      "70   taxon1    20   (22.222%)\t18\n",
      "71   taxon156  20   (22.222%)\t19\n",
      "72   taxon28   20   (22.222%)\t19\n",
      "73   taxon402  20   (22.222%)\t19\n",
      "74   taxon314  20   (22.222%)\t20\n",
      "75   taxon92   20   (22.222%)\t19\n",
      "76   taxon339  20   (22.222%)\t19\n",
      "77   taxon176  20   (22.222%)\t17\n",
      "78   taxon36   20   (22.222%)\t19\n",
      "79   taxon287  20   (22.222%)\t19\n",
      "80   taxon57   20   (22.222%)\t19\n",
      "81   taxon311  20   (22.222%)\t19\n",
      "82   taxon241  20   (22.222%)\t20\n",
      "83   taxon184  20   (22.222%)\t19\n",
      "84   taxon422  20   (22.222%)\t16\n",
      "85   taxon245  20   (22.222%)\t19\n",
      "86   taxon71   20   (22.222%)\t19\n",
      "87   taxon313  20   (22.222%)\t19\n",
      "88   taxon235  20   (22.222%)\t20\n",
      "89   taxon220  20   (22.222%)\t19\n",
      "90   taxon277  20   (22.222%)\t18\n",
      "91   taxon271  20   (22.222%)\t18\n",
      "92   taxon76   20   (22.222%)\t20\n",
      "93   taxon56   20   (22.222%)\t19\n",
      "94   taxon82   20   (22.222%)\t20\n",
      "95   taxon120  20   (22.222%)\t19\n",
      "96   taxon237  20   (22.222%)\t18\n",
      "97   taxon217  20   (22.222%)\t20\n",
      "98   taxon174  20   (22.222%)\t18\n",
      "99   taxon40   20   (22.222%)\t17\n",
      "100  taxon263  20   (22.222%)\t18\n",
      "101  taxon308  20   (22.222%)\t19\n",
      "102  taxon282  20   (22.222%)\t19\n",
      "103  taxon321  20   (22.222%)\t19\n",
      "104  taxon93   20   (22.222%)\t19\n",
      "105  taxon89   20   (22.222%)\t19\n",
      "106  taxon88   20   (22.222%)\t20\n",
      "107  taxon404  20   (22.222%)\t19\n",
      "108  taxon91   20   (22.222%)\t19\n",
      "109  taxon90   20   (22.222%)\t19\n",
      "110  taxon207  20   (22.222%)\t18\n",
      "111  taxon415  20   (22.222%)\t20\n",
      "112  taxon340  20   (22.222%)\t20\n",
      "113  taxon83   20   (22.222%)\t20\n",
      "114  taxon208  20   (22.222%)\t20\n",
      "115  taxon324  20   (22.222%)\t19\n",
      "116  taxon338  20   (22.222%)\t19\n",
      "117  taxon244  20   (22.222%)\t19\n",
      "118  taxon209  20   (22.222%)\t20\n",
      "119  taxon327  20   (22.222%)\t19\n",
      "120  taxon204  20   (22.222%)\t20\n",
      "121  taxon136  20   (22.222%)\t18\n",
      "122  taxon13   20   (22.222%)\t18\n",
      "123  taxon65   20   (22.222%)\t19\n",
      "124  taxon66   20   (22.222%)\t20\n",
      "125  taxon343  20   (22.222%)\t19\n",
      "126  taxon326  20   (22.222%)\t19\n",
      "127  taxon414  20   (22.222%)\t19\n",
      "128  taxon332  20   (22.222%)\t19\n",
      "129  taxon290  20   (22.222%)\t20\n",
      "130  taxon417  20   (22.222%)\t19\n",
      "131  taxon256  20   (22.222%)\t19\n",
      "132  taxon189  20   (22.222%)\t18\n",
      "133  taxon38   20   (22.222%)\t19\n",
      "134  taxon361  20   (22.222%)\t20\n",
      "135  taxon346  20   (22.222%)\t19\n",
      "136  taxon396  20   (22.222%)\t19\n",
      "137  taxon323  20   (22.222%)\t19\n",
      "138  taxon231  20   (22.222%)\t20\n",
      "139  taxon407  20   (22.222%)\t19\n",
      "140  taxon347  20   (22.222%)\t19\n",
      "141  taxon300  20   (22.222%)\t19\n",
      "142  taxon171  20   (22.222%)\t18\n",
      "143  taxon47   20   (22.222%)\t19\n",
      "144  taxon246  20   (22.222%)\t20\n",
      "145  taxon285  20   (22.222%)\t19\n",
      "146  taxon60   20   (22.222%)\t20\n",
      "147  taxon129  20   (22.222%)\t18\n",
      "148  taxon435  20   (22.222%)\t19\n",
      "149  taxon55   20   (22.222%)\t18\n",
      "150  taxon384  20   (22.222%)\t20\n",
      "151  taxon383  20   (22.222%)\t20\n",
      "152  taxon288  20   (22.222%)\t19\n",
      "153  taxon46   20   (22.222%)\t19\n",
      "154  taxon59   20   (22.222%)\t20\n",
      "155  taxon200  20   (22.222%)\t20\n",
      "156  taxon365  20   (22.222%)\t19\n",
      "157  taxon284  20   (22.222%)\t18\n",
      "158  taxon139  20   (22.222%)\t18\n",
      "159  taxon437  20   (22.222%)\t17\n",
      "160  taxon193  20   (22.222%)\t19\n",
      "161  taxon17   20   (22.222%)\t20\n",
      "162  taxon125  20   (22.222%)\t19\n",
      "163  taxon426  20   (22.222%)\t20\n",
      "164  taxon301  20   (22.222%)\t19\n",
      "165  taxon130  20   (22.222%)\t18\n",
      "166  taxon434  20   (22.222%)\t18\n",
      "167  taxon133  20   (22.222%)\t18\n",
      "168  taxon29   20   (22.222%)\t17\n",
      "169  taxon274  20   (22.222%)\t19\n",
      "170  taxon387  20   (22.222%)\t20\n",
      "171  taxon52   20   (22.222%)\t19\n",
      "172  taxon218  20   (22.222%)\t20\n",
      "173  taxon382  20   (22.222%)\t19\n",
      "174  taxon228  20   (22.222%)\t20\n",
      "175  taxon438  20   (22.222%)\t19\n",
      "176  taxon148  20   (22.222%)\t19\n",
      "177  taxon195  20   (22.222%)\t18\n",
      "178  taxon427  20   (22.222%)\t18\n",
      "179  taxon162  20   (22.222%)\t16\n",
      "180  taxon26   20   (22.222%)\t19\n",
      "181  taxon388  20   (22.222%)\t20\n",
      "182  taxon84   20   (22.222%)\t19\n",
      "183  taxon225  21   (23.333%)\t20\n",
      "184  taxon187  20   (22.222%)\t18\n",
      "185  taxon424  20   (22.222%)\t17\n",
      "186  taxon186  20   (22.222%)\t18\n",
      "187  taxon423  20   (22.222%)\t17\n",
      "188  taxon240  20   (22.222%)\t19\n",
      "189  taxon322  20   (22.222%)\t19\n",
      "190  taxon247  20   (22.222%)\t20\n",
      "191  taxon261  20   (22.222%)\t20\n",
      "192  taxon353  20   (22.222%)\t20\n",
      "193  taxon355  20   (22.222%)\t20\n",
      "194  taxon354  20   (22.222%)\t19\n",
      "195  taxon64   20   (22.222%)\t20\n",
      "196  taxon80   20   (22.222%)\t20\n",
      "197  taxon229  20   (22.222%)\t20\n",
      "198  taxon319  20   (22.222%)\t19\n",
      "199  taxon181  20   (22.222%)\t20\n",
      "200  taxon123  20   (22.222%)\t18\n",
      "201  taxon180  20   (22.222%)\t19\n",
      "202  taxon182  20   (22.222%)\t18\n",
      "203  taxon45   20   (22.222%)\t19\n",
      "204  taxon122  20   (22.222%)\t19\n",
      "205  taxon18   20   (22.222%)\t19\n",
      "206  taxon265  20   (22.222%)\t19\n",
      "207  taxon309  20   (22.222%)\t19\n",
      "208  taxon269  20   (22.222%)\t17\n",
      "209  taxon109  20   (22.222%)\t19\n",
      "210  taxon318  20   (22.222%)\t19\n",
      "211  taxon104  20   (22.222%)\t19\n",
      "212  taxon375  20   (22.222%)\t19\n",
      "213  taxon232  20   (22.222%)\t19\n",
      "214  taxon376  20   (22.222%)\t18\n",
      "215  taxon110  20   (22.222%)\t19\n",
      "216  taxon53   20   (22.222%)\t19\n",
      "217  taxon98   20   (22.222%)\t19\n",
      "218  taxon75   20   (22.222%)\t19\n",
      "219  taxon99   20   (22.222%)\t19\n",
      "220  taxon374  20   (22.222%)\t18\n",
      "221  taxon336  20   (22.222%)\t20\n",
      "222  taxon409  20   (22.222%)\t19\n",
      "223  taxon408  20   (22.222%)\t18\n",
      "224  taxon72   20   (22.222%)\t17\n",
      "225  taxon212  20   (22.222%)\t20\n",
      "226  taxon219  20   (22.222%)\t18\n",
      "227  taxon100  20   (22.222%)\t20\n",
      "228  taxon73   20   (22.222%)\t18\n",
      "229  taxon395  20   (22.222%)\t19\n",
      "230  taxon103  20   (22.222%)\t20\n",
      "231  taxon116  20   (22.222%)\t19\n",
      "232  taxon102  20   (22.222%)\t20\n",
      "233  taxon239  20   (22.222%)\t19\n",
      "234  taxon152  20   (22.222%)\t19\n",
      "235  taxon3    20   (22.222%)\t19\n",
      "236  taxon362  20   (22.222%)\t17\n",
      "237  taxon266  20   (22.222%)\t18\n",
      "238  taxon331  20   (22.222%)\t17\n",
      "239  taxon296  20   (22.222%)\t18\n",
      "240  taxon224  20   (22.222%)\t20\n",
      "241  taxon223  20   (22.222%)\t20\n",
      "242  taxon160  20   (22.222%)\t17\n",
      "243  taxon22   20   (22.222%)\t19\n",
      "244  taxon385  20   (22.222%)\t19\n",
      "245  taxon330  20   (22.222%)\t19\n",
      "246  taxon161  20   (22.222%)\t18\n",
      "247  taxon31   20   (22.222%)\t17\n",
      "248  taxon257  20   (22.222%)\t19\n",
      "249  taxon397  20   (22.222%)\t19\n",
      "250  taxon386  20   (22.222%)\t19\n",
      "251  taxon168  20   (22.222%)\t18\n",
      "252  taxon49   20   (22.222%)\t18\n",
      "253  taxon106  20   (22.222%)\t19\n",
      "254  taxon360  20   (22.222%)\t19\n",
      "255  taxon312  20   (22.222%)\t19\n",
      "256  taxon78   20   (22.222%)\t19\n",
      "257  taxon94   20   (22.222%)\t19\n",
      "258  taxon113  20   (22.222%)\t20\n",
      "259  taxon115  20   (22.222%)\t19\n",
      "260  taxon211  20   (22.222%)\t20\n",
      "261  taxon303  20   (22.222%)\t19\n",
      "262  taxon112  20   (22.222%)\t20\n",
      "263  taxon359  20   (22.222%)\t19\n",
      "264  taxon97   20   (22.222%)\t19\n",
      "265  taxon149  20   (22.222%)\t19\n",
      "266  taxon8    20   (22.222%)\t19\n",
      "267  taxon114  20   (22.222%)\t20\n",
      "268  taxon96   20   (22.222%)\t20\n",
      "269  taxon421  20   (22.222%)\t20\n",
      "270  taxon368  20   (22.222%)\t19\n",
      "271  taxon20   20   (22.222%)\t19\n",
      "272  taxon74   20   (22.222%)\t20\n",
      "273  taxon297  20   (22.222%)\t19\n",
      "274  taxon77   20   (22.222%)\t18\n",
      "275  taxon105  20   (22.222%)\t19\n",
      "276  taxon111  20   (22.222%)\t20\n",
      "277  taxon398  20   (22.222%)\t19\n",
      "278  taxon268  20   (22.222%)\t18\n",
      "279  taxon363  20   (22.222%)\t18\n",
      "280  taxon69   20   (22.222%)\t19\n",
      "281  taxon267  20   (22.222%)\t17\n",
      "282  taxon153  20   (22.222%)\t19\n",
      "283  taxon5    20   (22.222%)\t19\n",
      "284  taxon295  20   (22.222%)\t19\n",
      "285  taxon242  20   (22.222%)\t19\n",
      "286  taxon315  20   (22.222%)\t19\n",
      "287  taxon70   20   (22.222%)\t20\n",
      "288  taxon119  20   (22.222%)\t19\n",
      "289  taxon280  20   (22.222%)\t19\n",
      "290  taxon273  20   (22.222%)\t18\n",
      "291  taxon79   20   (22.222%)\t18\n",
      "292  taxon234  20   (22.222%)\t20\n",
      "293  taxon394  20   (22.222%)\t19\n",
      "294  taxon167  20   (22.222%)\t18\n",
      "295  taxon42   20   (22.222%)\t18\n",
      "296  taxon107  20   (22.222%)\t19\n",
      "297  taxon101  20   (22.222%)\t19\n",
      "298  taxon131  20   (22.222%)\t18\n",
      "299  taxon430  20   (22.222%)\t18\n",
      "300  taxon166  20   (22.222%)\t18\n",
      "301  taxon41   20   (22.222%)\t18\n",
      "302  taxon194  20   (22.222%)\t17\n",
      "303  taxon21   20   (22.222%)\t20\n",
      "304  taxon298  20   (22.222%)\t19\n",
      "305  taxon159  20   (22.222%)\t18\n",
      "306  taxon335  20   (22.222%)\t19\n",
      "307  taxon169  20   (22.222%)\t18\n",
      "308  taxon50   20   (22.222%)\t17\n",
      "309  taxon170  20   (22.222%)\t18\n",
      "310  taxon48   20   (22.222%)\t20\n",
      "311  taxon307  20   (22.222%)\t19\n",
      "312  taxon341  20   (22.222%)\t18\n",
      "313  taxon286  20   (22.222%)\t19\n",
      "314  taxon349  20   (22.222%)\t18\n",
      "315  taxon344  20   (22.222%)\t19\n",
      "316  taxon413  20   (22.222%)\t18\n",
      "317  taxon294  20   (22.222%)\t19\n",
      "318  taxon259  20   (22.222%)\t20\n",
      "319  taxon81   20   (22.222%)\t20\n",
      "320  taxon393  20   (22.222%)\t20\n",
      "321  taxon188  20   (22.222%)\t20\n",
      "322  taxon419  20   (22.222%)\t18\n",
      "323  taxon302  20   (22.222%)\t19\n",
      "324  taxon137  20   (22.222%)\t18\n",
      "325  taxon63   20   (22.222%)\t19\n",
      "326  taxon51   20   (22.222%)\t19\n",
      "327  taxon270  20   (22.222%)\t17\n",
      "328  taxon406  20   (22.222%)\t19\n",
      "329  taxon272  20   (22.222%)\t18\n",
      "330  taxon416  20   (22.222%)\t18\n",
      "331  taxon213  20   (22.222%)\t20\n",
      "332  taxon191  20   (22.222%)\t19\n",
      "333  taxon16   20   (22.222%)\t17\n",
      "334  taxon154  20   (22.222%)\t18\n",
      "335  taxon275  20   (22.222%)\t19\n",
      "336  taxon367  20   (22.222%)\t19\n",
      "337  taxon243  20   (22.222%)\t20\n",
      "338  taxon328  20   (22.222%)\t19\n",
      "339  taxon236  20   (22.222%)\t19\n",
      "340  taxon226  20   (22.222%)\t20\n",
      "341  taxon230  20   (22.222%)\t20\n",
      "342  taxon255  20   (22.222%)\t19\n",
      "343  taxon163  20   (22.222%)\t18\n",
      "344  taxon30   20   (22.222%)\t17\n",
      "345  taxon205  20   (22.222%)\t19\n",
      "346  taxon86   20   (22.222%)\t19\n",
      "347  taxon320  20   (22.222%)\t19\n",
      "348  taxon85   20   (22.222%)\t20\n",
      "349  taxon178  20   (22.222%)\t18\n",
      "350  taxon37   20   (22.222%)\t17\n",
      "351  taxon155  20   (22.222%)\t18\n",
      "352  taxon439  20   (22.222%)\t18\n",
      "353  taxon140  20   (22.222%)\t18\n",
      "354  taxon2    20   (22.222%)\t18\n",
      "355  taxon433  20   (22.222%)\t20\n",
      "356  taxon202  20   (22.222%)\t20\n",
      "357  taxon62   20   (22.222%)\t20\n",
      "358  taxon350  20   (22.222%)\t19\n",
      "359  taxon378  20   (22.222%)\t19\n",
      "360  taxon252  20   (22.222%)\t20\n",
      "361  taxon289  20   (22.222%)\t19\n",
      "362  taxon391  20   (22.222%)\t19\n",
      "363  taxon87   20   (22.222%)\t19\n",
      "364  taxon215  20   (22.222%)\t20\n",
      "365  taxon337  20   (22.222%)\t19\n",
      "366  taxon32   20   (22.222%)\t18\n",
      "367  taxon251  20   (22.222%)\t20\n",
      "368  taxon381  20   (22.222%)\t19\n",
      "369  taxon358  20   (22.222%)\t18\n",
      "370  taxon142  20   (22.222%)\t18\n",
      "371  taxon11   20   (22.222%)\t18\n",
      "372  taxon310  20   (22.222%)\t19\n",
      "373  taxon352  20   (22.222%)\t19\n",
      "374  taxon222  20   (22.222%)\t19\n",
      "375  taxon138  20   (22.222%)\t18\n",
      "376  taxon14   20   (22.222%)\t20\n",
      "377  taxon198  20   (22.222%)\t19\n",
      "378  taxon134  20   (22.222%)\t18\n",
      "379  taxon33   20   (22.222%)\t17\n",
      "380  taxon278  20   (22.222%)\t19\n",
      "381  taxon143  20   (22.222%)\t19\n",
      "382  taxon10   20   (22.222%)\t18\n",
      "383  taxon175  20   (22.222%)\t18\n",
      "384  taxon39   20   (22.222%)\t19\n",
      "385  taxon61   20   (22.222%)\t20\n",
      "386  taxon292  20   (22.222%)\t20\n",
      "387  taxon412  20   (22.222%)\t19\n",
      "388  taxon196  20   (22.222%)\t19\n",
      "389  taxon201  20   (22.222%)\t20\n",
      "390  taxon345  20   (22.222%)\t19\n",
      "391  taxon127  20   (22.222%)\t18\n",
      "392  taxon431  20   (22.222%)\t19\n",
      "393  taxon304  20   (22.222%)\t19\n",
      "394  taxon351  20   (22.222%)\t19\n",
      "395  taxon377  20   (22.222%)\t18\n",
      "396  taxon197  20   (22.222%)\t19\n",
      "397  taxon121  20   (22.222%)\t19\n",
      "398  taxon216  20   (22.222%)\t18\n",
      "399  taxon203  20   (22.222%)\t20\n",
      "400  taxon389  20   (22.222%)\t19\n",
      "401  taxon390  20   (22.222%)\t20\n",
      "402  taxon147  20   (22.222%)\t19\n",
      "403  taxon4    20   (22.222%)\t18\n",
      "404  taxon185  20   (22.222%)\t18\n",
      "405  taxon7    20   (22.222%)\t18\n",
      "406  taxon379  20   (22.222%)\t19\n",
      "407  taxon293  20   (22.222%)\t19\n",
      "408  taxon250  20   (22.222%)\t20\n",
      "409  taxon141  20   (22.222%)\t18\n",
      "410  taxon12   20   (22.222%)\t19\n",
      "411  taxon411  20   (22.222%)\t19\n",
      "412  taxon317  20   (22.222%)\t19\n",
      "413  taxon199  20   (22.222%)\t19\n",
      "414  taxon158  20   (22.222%)\t18\n",
      "415  taxon58   20   (22.222%)\t19\n",
      "416  taxon291  20   (22.222%)\t20\n",
      "417  taxon144  20   (22.222%)\t19\n",
      "418  taxon9    20   (22.222%)\t18\n",
      "419  taxon380  20   (22.222%)\t19\n",
      "420  taxon214  20   (22.222%)\t20\n",
      "421  taxon128  20   (22.222%)\t18\n",
      "422  taxon432  20   (22.222%)\t19\n",
      "423  taxon325  20   (22.222%)\t19\n",
      "424  taxon249  20   (22.222%)\t20\n",
      "425  taxon400  20   (22.222%)\t20\n",
      "426  taxon276  20   (22.222%)\t19\n",
      "427  taxon172  20   (22.222%)\t18\n",
      "428  taxon43   20   (22.222%)\t18\n",
      "429  taxon342  20   (22.222%)\t19\n",
      "430  taxon410  20   (22.222%)\t19\n",
      "431  taxon248  20   (22.222%)\t19\n",
      "432  taxon173  20   (22.222%)\t18\n",
      "433  taxon44   20   (22.222%)\t18\n",
      "**** TOTAL     8667 (22.240%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.002 seconds\n",
      "Parsimony score: 4691\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15861_1/15861_1.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 26.939 MB RAM is required!\n",
      "Current best tree score: 4691\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "NOTE: REPS computation is segmented into 2 parts for speed...\n",
      "segment#1 ends at 48\n",
      "segment#2 ends at 86\n",
      "0 seconds.\n",
      "Current best score: 4608 / CPU time: 21\n",
      "\n",
      "Time: 0h:0m:20s\n",
      "BETTER TREE FOUND at iteration 2: 4594 / CPU time: 27s\n",
      "BETTER TREE FOUND at iteration 4: 4593 / CPU time: 30s\n",
      "Completed MPBOOT for: 15861_1\n",
      "##########################\n",
      "Running MPBOOT for: 20239_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/20239_0/20239_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/20239_0/20239_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/20239_0/20239_0\n",
      "Seed:    336894 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:30 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20239_0/20239_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "WARNING: 2 sites contain only gaps or ambiguous chars.\n",
      "Alignment has 320 sequences with 1499 columns and 768 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon113  2    (0.133%)\t4\n",
      "2    taxon61   2    (0.133%)\t4\n",
      "3    taxon309  2    (0.133%)\t4\n",
      "4    taxon69   2    (0.133%)\t4\n",
      "5    taxon186  2    (0.133%)\t4\n",
      "6    taxon175  2    (0.133%)\t4\n",
      "7    taxon235  2    (0.133%)\t4\n",
      "8    taxon157  2    (0.133%)\t4\n",
      "9    taxon146  2    (0.133%)\t4\n",
      "10   taxon193  2    (0.133%)\t4\n",
      "11   taxon194  2    (0.133%)\t4\n",
      "12   taxon66   2    (0.133%)\t4\n",
      "13   taxon170  2    (0.133%)\t4\n",
      "14   taxon233  2    (0.133%)\t4\n",
      "15   taxon48   2    (0.133%)\t4\n",
      "16   taxon155  2    (0.133%)\t4\n",
      "17   taxon52   2    (0.133%)\t4\n",
      "18   taxon243  2    (0.133%)\t4\n",
      "19   taxon57   2    (0.133%)\t4\n",
      "20   taxon182  3    (0.200%)\t5\n",
      "21   taxon236  2    (0.133%)\t4\n",
      "22   taxon238  2    (0.133%)\t4\n",
      "23   taxon59   2    (0.133%)\t4\n",
      "24   taxon60   2    (0.133%)\t4\n",
      "25   taxon158  2    (0.133%)\t4\n",
      "26   taxon123  2    (0.133%)\t4\n",
      "27   taxon56   2    (0.133%)\t4\n",
      "28   taxon124  2    (0.133%)\t4\n",
      "29   taxon237  3    (0.200%)\t4\n",
      "30   taxon156  2    (0.133%)\t4\n",
      "31   taxon122  2    (0.133%)\t4\n",
      "32   taxon305  2    (0.133%)\t4\n",
      "33   taxon244  2    (0.133%)\t4\n",
      "34   taxon179  2    (0.133%)\t4\n",
      "35   taxon180  2    (0.133%)\t4\n",
      "36   taxon104  2    (0.133%)\t4\n",
      "37   taxon112  2    (0.133%)\t4\n",
      "38   taxon317  2    (0.133%)\t4\n",
      "39   taxon114  2    (0.133%)\t4\n",
      "40   taxon51   2    (0.133%)\t4\n",
      "41   taxon109  2    (0.133%)\t4\n",
      "42   taxon245  2    (0.133%)\t4\n",
      "43   taxon246  2    (0.133%)\t4\n",
      "44   taxon50   3    (0.200%)\t5\n",
      "45   taxon55   2    (0.133%)\t4\n",
      "46   taxon54   2    (0.133%)\t4\n",
      "47   taxon202  2    (0.133%)\t4\n",
      "48   taxon229  2    (0.133%)\t4\n",
      "49   taxon162  2    (0.133%)\t4\n",
      "50   taxon110  3    (0.200%)\t5\n",
      "51   taxon163  2    (0.133%)\t4\n",
      "52   taxon115  2    (0.133%)\t4\n",
      "53   taxon116  2    (0.133%)\t4\n",
      "54   taxon169  2    (0.133%)\t4\n",
      "55   taxon192  2    (0.133%)\t4\n",
      "56   taxon221  2    (0.133%)\t4\n",
      "57   taxon240  2    (0.133%)\t4\n",
      "58   taxon198  2    (0.133%)\t4\n",
      "59   taxon58   2    (0.133%)\t4\n",
      "60   taxon147  2    (0.133%)\t4\n",
      "61   taxon216  2    (0.133%)\t4\n",
      "62   taxon111  2    (0.133%)\t4\n",
      "63   taxon142  2    (0.133%)\t4\n",
      "64   taxon278  2    (0.133%)\t4\n",
      "65   taxon210  2    (0.133%)\t4\n",
      "66   taxon307  2    (0.133%)\t4\n",
      "67   taxon183  2    (0.133%)\t4\n",
      "68   taxon136  2    (0.133%)\t4\n",
      "69   taxon107  2    (0.133%)\t4\n",
      "70   taxon106  2    (0.133%)\t4\n",
      "71   taxon185  2    (0.133%)\t4\n",
      "72   taxon174  2    (0.133%)\t4\n",
      "73   taxon228  2    (0.133%)\t4\n",
      "74   taxon139  2    (0.133%)\t4\n",
      "75   taxon131  2    (0.133%)\t4\n",
      "76   taxon130  2    (0.133%)\t4\n",
      "77   taxon117  2    (0.133%)\t4\n",
      "78   taxon120  2    (0.133%)\t4\n",
      "79   taxon119  2    (0.133%)\t4\n",
      "80   taxon150  2    (0.133%)\t4\n",
      "81   taxon145  2    (0.133%)\t4\n",
      "82   taxon144  2    (0.133%)\t4\n",
      "83   taxon46   2    (0.133%)\t4\n",
      "84   taxon199  2    (0.133%)\t4\n",
      "85   taxon190  3    (0.200%)\t5\n",
      "86   taxon224  2    (0.133%)\t4\n",
      "87   taxon154  2    (0.133%)\t4\n",
      "88   taxon121  2    (0.133%)\t4\n",
      "89   taxon105  2    (0.133%)\t4\n",
      "90   taxon140  2    (0.133%)\t4\n",
      "91   taxon132  2    (0.133%)\t4\n",
      "92   taxon249  2    (0.133%)\t4\n",
      "93   taxon148  2    (0.133%)\t4\n",
      "94   taxon188  2    (0.133%)\t4\n",
      "95   taxon171  2    (0.133%)\t4\n",
      "96   taxon177  2    (0.133%)\t4\n",
      "97   taxon47   2    (0.133%)\t4\n",
      "98   taxon220  2    (0.133%)\t4\n",
      "99   taxon232  2    (0.133%)\t4\n",
      "100  taxon151  2    (0.133%)\t4\n",
      "101  taxon152  2    (0.133%)\t4\n",
      "102  taxon125  2    (0.133%)\t4\n",
      "103  taxon133  2    (0.133%)\t4\n",
      "104  taxon126  2    (0.133%)\t4\n",
      "105  taxon127  2    (0.133%)\t4\n",
      "106  taxon223  2    (0.133%)\t4\n",
      "107  taxon172  2    (0.133%)\t4\n",
      "108  taxon165  2    (0.133%)\t4\n",
      "109  taxon167  2    (0.133%)\t4\n",
      "110  taxon164  2    (0.133%)\t4\n",
      "111  taxon168  2    (0.133%)\t4\n",
      "112  taxon218  2    (0.133%)\t4\n",
      "113  taxon149  2    (0.133%)\t4\n",
      "114  taxon141  2    (0.133%)\t4\n",
      "115  taxon118  2    (0.133%)\t4\n",
      "116  taxon134  2    (0.133%)\t4\n",
      "117  taxon2    2    (0.133%)\t4\n",
      "118  taxon187  2    (0.133%)\t4\n",
      "119  taxon176  2    (0.133%)\t4\n",
      "120  taxon138  2    (0.133%)\t4\n",
      "121  taxon40   2    (0.133%)\t4\n",
      "122  taxon41   2    (0.133%)\t4\n",
      "123  taxon261  2    (0.133%)\t4\n",
      "124  taxon259  2    (0.133%)\t4\n",
      "125  taxon100  2    (0.133%)\t4\n",
      "126  taxon265  2    (0.133%)\t4\n",
      "127  taxon262  2    (0.133%)\t4\n",
      "128  taxon257  2    (0.133%)\t4\n",
      "129  taxon99   2    (0.133%)\t4\n",
      "130  taxon258  2    (0.133%)\t4\n",
      "131  taxon274  2    (0.133%)\t4\n",
      "132  taxon266  2    (0.133%)\t4\n",
      "133  taxon263  2    (0.133%)\t4\n",
      "134  taxon283  2    (0.133%)\t4\n",
      "135  taxon98   2    (0.133%)\t4\n",
      "136  taxon256  2    (0.133%)\t4\n",
      "137  taxon260  2    (0.133%)\t4\n",
      "138  taxon159  2    (0.133%)\t4\n",
      "139  taxon178  2    (0.133%)\t4\n",
      "140  taxon73   2    (0.133%)\t4\n",
      "141  taxon275  2    (0.133%)\t4\n",
      "142  taxon276  2    (0.133%)\t4\n",
      "143  taxon80   2    (0.133%)\t4\n",
      "144  taxon81   2    (0.133%)\t4\n",
      "145  taxon102  2    (0.133%)\t4\n",
      "146  taxon10   2    (0.133%)\t4\n",
      "147  taxon53   2    (0.133%)\t4\n",
      "148  taxon42   2    (0.133%)\t4\n",
      "149  taxon319  2    (0.133%)\t4\n",
      "150  taxon12   2    (0.133%)\t4\n",
      "151  taxon253  2    (0.133%)\t4\n",
      "152  taxon103  2    (0.133%)\t4\n",
      "153  taxon11   2    (0.133%)\t4\n",
      "154  taxon318  2    (0.133%)\t4\n",
      "155  taxon255  2    (0.133%)\t4\n",
      "156  taxon94   2    (0.133%)\t4\n",
      "157  taxon91   2    (0.133%)\t4\n",
      "158  taxon214  2    (0.133%)\t4\n",
      "159  taxon304  2    (0.133%)\t4\n",
      "160  taxon44   2    (0.133%)\t4\n",
      "161  taxon303  2    (0.133%)\t4\n",
      "162  taxon213  2    (0.133%)\t4\n",
      "163  taxon34   2    (0.133%)\t4\n",
      "164  taxon97   2    (0.133%)\t4\n",
      "165  taxon21   2    (0.133%)\t4\n",
      "166  taxon93   2    (0.133%)\t4\n",
      "167  taxon3    2    (0.133%)\t4\n",
      "168  taxon89   2    (0.133%)\t4\n",
      "169  taxon310  2    (0.133%)\t4\n",
      "170  taxon284  2    (0.133%)\t4\n",
      "171  taxon13   2    (0.133%)\t4\n",
      "172  taxon292  2    (0.133%)\t4\n",
      "173  taxon311  2    (0.133%)\t4\n",
      "174  taxon31   2    (0.133%)\t4\n",
      "175  taxon222  2    (0.133%)\t4\n",
      "176  taxon227  2    (0.133%)\t4\n",
      "177  taxon32   2    (0.133%)\t4\n",
      "178  taxon231  2    (0.133%)\t4\n",
      "179  taxon181  2    (0.133%)\t4\n",
      "180  taxon197  2    (0.133%)\t4\n",
      "181  taxon196  2    (0.133%)\t4\n",
      "182  taxon76   2    (0.133%)\t4\n",
      "183  taxon15   2    (0.133%)\t4\n",
      "184  taxon8    2    (0.133%)\t4\n",
      "185  taxon217  2    (0.133%)\t4\n",
      "186  taxon191  2    (0.133%)\t4\n",
      "187  taxon230  2    (0.133%)\t4\n",
      "188  taxon23   2    (0.133%)\t4\n",
      "189  taxon77   2    (0.133%)\t4\n",
      "190  taxon6    2    (0.133%)\t4\n",
      "191  taxon39   2    (0.133%)\t4\n",
      "192  taxon161  2    (0.133%)\t4\n",
      "193  taxon200  2    (0.133%)\t4\n",
      "194  taxon320  2    (0.133%)\t4\n",
      "195  taxon291  2    (0.133%)\t4\n",
      "196  taxon206  2    (0.133%)\t4\n",
      "197  taxon316  2    (0.133%)\t4\n",
      "198  taxon17   3    (0.200%)\t5\n",
      "199  taxon308  4    (0.267%)\t5\n",
      "200  taxon306  2    (0.133%)\t4\n",
      "201  taxon74   2    (0.133%)\t4\n",
      "202  taxon287  2    (0.133%)\t4\n",
      "203  taxon72   2    (0.133%)\t4\n",
      "204  taxon315  2    (0.133%)\t4\n",
      "205  taxon296  2    (0.133%)\t4\n",
      "206  taxon129  2    (0.133%)\t4\n",
      "207  taxon207  2    (0.133%)\t4\n",
      "208  taxon204  2    (0.133%)\t4\n",
      "209  taxon82   2    (0.133%)\t4\n",
      "210  taxon280  2    (0.133%)\t4\n",
      "211  taxon211  2    (0.133%)\t4\n",
      "212  taxon84   2    (0.133%)\t4\n",
      "213  taxon298  2    (0.133%)\t4\n",
      "214  taxon290  2    (0.133%)\t4\n",
      "215  taxon9    2    (0.133%)\t4\n",
      "216  taxon297  2    (0.133%)\t4\n",
      "217  taxon70   2    (0.133%)\t4\n",
      "218  taxon299  2    (0.133%)\t4\n",
      "219  taxon288  2    (0.133%)\t4\n",
      "220  taxon295  2    (0.133%)\t4\n",
      "221  taxon22   2    (0.133%)\t4\n",
      "222  taxon314  2    (0.133%)\t4\n",
      "223  taxon313  2    (0.133%)\t4\n",
      "224  taxon173  2    (0.133%)\t4\n",
      "225  taxon294  2    (0.133%)\t4\n",
      "226  taxon16   2    (0.133%)\t4\n",
      "227  taxon248  2    (0.133%)\t4\n",
      "228  taxon282  8    (0.534%)\t7\n",
      "229  taxon143  2    (0.133%)\t4\n",
      "230  taxon271  2    (0.133%)\t4\n",
      "231  taxon14   2    (0.133%)\t4\n",
      "232  taxon247  2    (0.133%)\t4\n",
      "233  taxon25   2    (0.133%)\t4\n",
      "234  taxon26   2    (0.133%)\t4\n",
      "235  taxon29   2    (0.133%)\t4\n",
      "236  taxon63   2    (0.133%)\t4\n",
      "237  taxon35   2    (0.133%)\t4\n",
      "238  taxon269  2    (0.133%)\t4\n",
      "239  taxon24   2    (0.133%)\t4\n",
      "240  taxon30   2    (0.133%)\t4\n",
      "241  taxon267  2    (0.133%)\t4\n",
      "242  taxon264  2    (0.133%)\t4\n",
      "243  taxon85   2    (0.133%)\t4\n",
      "244  taxon33   2    (0.133%)\t4\n",
      "245  taxon252  2    (0.133%)\t4\n",
      "246  taxon28   2    (0.133%)\t4\n",
      "247  taxon153  2    (0.133%)\t4\n",
      "248  taxon75   2    (0.133%)\t4\n",
      "249  taxon71   2    (0.133%)\t4\n",
      "250  taxon4    2    (0.133%)\t4\n",
      "251  taxon7    2    (0.133%)\t4\n",
      "252  taxon219  2    (0.133%)\t4\n",
      "253  taxon273  2    (0.133%)\t4\n",
      "254  taxon281  2    (0.133%)\t4\n",
      "255  taxon277  2    (0.133%)\t4\n",
      "256  taxon272  2    (0.133%)\t4\n",
      "257  taxon279  2    (0.133%)\t4\n",
      "258  taxon108  2    (0.133%)\t4\n",
      "259  taxon19   2    (0.133%)\t4\n",
      "260  taxon312  2    (0.133%)\t4\n",
      "261  taxon96   2    (0.133%)\t4\n",
      "262  taxon101  2    (0.133%)\t4\n",
      "263  taxon64   2    (0.133%)\t4\n",
      "264  taxon78   2    (0.133%)\t4\n",
      "265  taxon270  2    (0.133%)\t4\n",
      "266  taxon166  2    (0.133%)\t4\n",
      "267  taxon195  2    (0.133%)\t4\n",
      "268  taxon203  2    (0.133%)\t4\n",
      "269  taxon201  2    (0.133%)\t4\n",
      "270  taxon215  2    (0.133%)\t4\n",
      "271  taxon225  2    (0.133%)\t4\n",
      "272  taxon189  2    (0.133%)\t4\n",
      "273  taxon67   2    (0.133%)\t4\n",
      "274  taxon302  2    (0.133%)\t4\n",
      "275  taxon1    2    (0.133%)\t4\n",
      "276  taxon43   2    (0.133%)\t4\n",
      "277  taxon300  2    (0.133%)\t4\n",
      "278  taxon37   2    (0.133%)\t4\n",
      "279  taxon90   2    (0.133%)\t4\n",
      "280  taxon38   2    (0.133%)\t4\n",
      "281  taxon251  2    (0.133%)\t4\n",
      "282  taxon65   2    (0.133%)\t4\n",
      "283  taxon242  2    (0.133%)\t4\n",
      "284  taxon241  2    (0.133%)\t4\n",
      "285  taxon184  2    (0.133%)\t4\n",
      "286  taxon49   2    (0.133%)\t4\n",
      "287  taxon135  2    (0.133%)\t4\n",
      "288  taxon137  2    (0.133%)\t4\n",
      "289  taxon79   2    (0.133%)\t4\n",
      "290  taxon95   2    (0.133%)\t4\n",
      "291  taxon160  2    (0.133%)\t4\n",
      "292  taxon128  2    (0.133%)\t4\n",
      "293  taxon45   2    (0.133%)\t4\n",
      "294  taxon68   2    (0.133%)\t4\n",
      "295  taxon212  2    (0.133%)\t4\n",
      "296  taxon205  2    (0.133%)\t4\n",
      "297  taxon234  2    (0.133%)\t4\n",
      "298  taxon250  2    (0.133%)\t4\n",
      "299  taxon36   2    (0.133%)\t4\n",
      "300  taxon92   2    (0.133%)\t4\n",
      "301  taxon289  2    (0.133%)\t4\n",
      "302  taxon301  2    (0.133%)\t4\n",
      "303  taxon5    2    (0.133%)\t4\n",
      "304  taxon208  2    (0.133%)\t4\n",
      "305  taxon226  2    (0.133%)\t4\n",
      "306  taxon239  3    (0.200%)\t5\n",
      "307  taxon209  2    (0.133%)\t4\n",
      "308  taxon83   2    (0.133%)\t4\n",
      "309  taxon88   2    (0.133%)\t4\n",
      "310  taxon87   2    (0.133%)\t4\n",
      "311  taxon268  2    (0.133%)\t4\n",
      "312  taxon254  2    (0.133%)\t4\n",
      "313  taxon62   2    (0.133%)\t4\n",
      "314  taxon27   2    (0.133%)\t4\n",
      "315  taxon86   2    (0.133%)\t4\n",
      "316  taxon285  3    (0.200%)\t5\n",
      "317  taxon293  2    (0.133%)\t4\n",
      "318  taxon20   2    (0.133%)\t4\n",
      "319  taxon286  2    (0.133%)\t4\n",
      "320  taxon18   2    (0.133%)\t4\n",
      "**** TOTAL     656 (0.137%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.001 seconds\n",
      "Parsimony score: 4742\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.001 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20239_0/20239_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 43.281 MB RAM is required!\n",
      "Current best tree score: 4742\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "NOTE: REPS computation is segmented into 2 parts for speed...\n",
      "segment#1 ends at 416\n",
      "segment#2 ends at 754\n",
      "0 seconds.\n",
      "Completed MPBOOT for: 20239_0\n",
      "##########################\n",
      "Running MPBOOT for: 15631_6\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6\n",
      "Seed:    607176 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:30 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 9 sequences with 509 columns and 39 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon4   17   (3.340%)\t4\n",
      "2    taxon8   1    (0.196%)\t4\n",
      "3    taxon6   1    (0.196%)\t4\n",
      "4    taxon22  1    (0.196%)\t4\n",
      "5    taxon13  1    (0.196%)\t4\n",
      "6    taxon15  1    (0.196%)\t4\n",
      "7    taxon3   2    (0.393%)\t4\n",
      "8    taxon1   12   (2.358%)\t4\n",
      "9    taxon2   4    (0.786%)\t4\n",
      "**** TOTAL    40 (0.873%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 31\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.058 MB RAM is required!\n",
      "Current best tree score: 31\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 31 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 31 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 30 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 40 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 50 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 31 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  0 sec.\n",
      "NOTE: At the end, 3461 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "31 31 31 31 \n",
      "BEST SCORE FOUND : 31\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.splits.nex\n",
      "\n",
      "CPU time used for tree search: 0.245 sec (0h:0m:0s)\n",
      "Wall-clock time used for tree search: 0.245 sec (0h:0m:0s)\n",
      "Total CPU time used: 0.247 sec (0h:0m:0s)\n",
      "Total wall-clock time used: 0.247 sec (0h:0m:0s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.splits.nex...\n",
      "9 taxa and 64 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.contree\n",
      "Parsimony score of consensus tree: 31\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/15631_6/15631_6.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:35:30 2024\n",
      "Completed MPBOOT for: 15631_6\n",
      "##########################\n",
      "Running MPBOOT for: 20497_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0\n",
      "Seed:    856797 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:30 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 21 sequences with 419 columns and 149 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon30  17   (4.057%)\t4\n",
      "2    taxon29  16   (3.819%)\t4\n",
      "3    taxon34  18   (4.296%)\t4\n",
      "4    taxon33  10   (2.387%)\t4\n",
      "5    taxon35  20   (4.773%)\t4\n",
      "6    taxon28  13   (3.103%)\t4\n",
      "7    taxon26  11   (2.625%)\t4\n",
      "8    taxon5   9    (2.148%)\t4\n",
      "9    taxon36  11   (2.625%)\t4\n",
      "10   taxon32  16   (3.819%)\t4\n",
      "11   taxon25  17   (4.057%)\t4\n",
      "12   taxon31  18   (4.296%)\t4\n",
      "13   taxon1   8    (1.909%)\t4\n",
      "14   taxon7   26   (6.205%)\t4\n",
      "15   taxon21  8    (1.909%)\t4\n",
      "16   taxon24  8    (1.909%)\t4\n",
      "17   taxon3   7    (1.671%)\t4\n",
      "18   taxon6   8    (1.909%)\t4\n",
      "19   taxon9   26   (6.205%)\t4\n",
      "20   taxon27  13   (3.103%)\t4\n",
      "21   taxon4   15   (3.580%)\t4\n",
      "**** TOTAL    295 (3.353%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 195\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.532 MB RAM is required!\n",
      "Current best tree score: 195\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 194 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "BETTER TREE FOUND at iteration 1: 193 / CPU time: 0s\n",
      "\n",
      "RATCHET Iteration 10 / Score: 193 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 30 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 40 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 50 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 194 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 193 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  1 sec.\n",
      "NOTE: At the end, 8710 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "193 193 193 193 193 \n",
      "BEST SCORE FOUND : 193\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.splits.nex\n",
      "\n",
      "CPU time used for tree search: 0.851 sec (0h:0m:0s)\n",
      "Wall-clock time used for tree search: 0.851 sec (0h:0m:0s)\n",
      "Total CPU time used: 0.862 sec (0h:0m:0s)\n",
      "Total wall-clock time used: 0.863 sec (0h:0m:0s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.splits.nex...\n",
      "21 taxa and 264 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.contree\n",
      "Parsimony score of consensus tree: 194\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/20497_0/20497_0.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:35:31 2024\n",
      "Completed MPBOOT for: 20497_0\n",
      "##########################\n",
      "Running MPBOOT for: 18883_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/18883_0/18883_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/18883_0/18883_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/18883_0/18883_0\n",
      "Seed:    724488 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:31 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/18883_0/18883_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 26 sequences with 976 columns and 409 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon22  284  (29.098%)\t4\n",
      "2    taxon23  294  (30.123%)\t4\n",
      "3    taxon1   267  (27.357%)\t4\n",
      "4    taxon15  72   (7.377%)\t5\n",
      "5    taxon16  73   (7.480%)\t5\n",
      "6    taxon17  83   (8.504%)\t4\n",
      "7    taxon24  450  (46.107%)\t4\n",
      "8    taxon26  266  (27.254%)\t4\n",
      "9    taxon21  266  (27.254%)\t4\n",
      "10   taxon2   434  (44.467%)\t4\n",
      "11   taxon5   267  (27.357%)\t4\n",
      "12   taxon14  367  (37.602%)\t4\n",
      "13   taxon19  261  (26.742%)\t4\n",
      "14   taxon7   616  (63.115%) !!!\t4\n",
      "15   taxon4   249  (25.512%)\t4\n",
      "16   taxon3   242  (24.795%)\t4\n",
      "17   taxon20  244  (25.000%)\t4\n",
      "18   taxon18  87   (8.914%)\t5\n",
      "19   taxon13  84   (8.607%)\t4\n",
      "20   taxon11  528  (54.098%) !!!\t4\n",
      "21   taxon8   269  (27.561%)\t4\n",
      "22   taxon6   228  (23.361%)\t4\n",
      "23   taxon12  258  (26.434%)\t4\n",
      "24   taxon25  504  (51.639%) !!!\t4\n",
      "25   taxon10  258  (26.434%)\t4\n",
      "26   taxon9   258  (26.434%)\t4\n",
      "WARNING: 3 sequences contain more than 50% gaps/ambiguity\n",
      "**** TOTAL    7209 (28.409%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 409\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/18883_0/18883_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 1.791 MB RAM is required!\n",
      "Current best tree score: 409\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 406 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 406 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 406 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "RATCHET Iteration 30 / Score: 406 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Completed MPBOOT for: 18883_0\n",
      "##########################\n",
      "Running MPBOOT for: 10214_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0\n",
      "Seed:    85156 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:32 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 37 sequences with 809 columns and 287 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon2   23   (2.843%)\t4\n",
      "2    taxon35  24   (2.967%)\t4\n",
      "3    taxon39  23   (2.843%)\t4\n",
      "4    taxon5   14   (1.731%)\t4\n",
      "5    taxon36  32   (3.956%)\t4\n",
      "6    taxon29  9    (1.112%)\t4\n",
      "7    taxon21  11   (1.360%)\t4\n",
      "8    taxon1   23   (2.843%)\t4\n",
      "9    taxon34  23   (2.843%)\t4\n",
      "10   taxon41  24   (2.967%)\t4\n",
      "11   taxon3   14   (1.731%)\t4\n",
      "12   taxon20  8    (0.989%)\t4\n",
      "13   taxon8   15   (1.854%)\t4\n",
      "14   taxon23  11   (1.360%)\t4\n",
      "15   taxon4   15   (1.854%)\t4\n",
      "16   taxon33  23   (2.843%)\t4\n",
      "17   taxon18  21   (2.596%)\t4\n",
      "18   taxon22  11   (1.360%)\t4\n",
      "19   taxon40  24   (2.967%)\t4\n",
      "20   taxon14  9    (1.112%)\t4\n",
      "21   taxon16  11   (1.360%)\t4\n",
      "22   taxon13  10   (1.236%)\t4\n",
      "23   taxon12  13   (1.607%)\t4\n",
      "24   taxon7   16   (1.978%)\t4\n",
      "25   taxon38  24   (2.967%)\t4\n",
      "26   taxon32  23   (2.843%)\t4\n",
      "27   taxon17  12   (1.483%)\t4\n",
      "28   taxon37  23   (2.843%)\t4\n",
      "29   taxon10  18   (2.225%)\t4\n",
      "30   taxon9   16   (1.978%)\t4\n",
      "31   taxon11  13   (1.607%)\t4\n",
      "32   taxon25  11   (1.360%)\t4\n",
      "33   taxon42  23   (2.843%)\t4\n",
      "34   taxon19  17   (2.101%)\t4\n",
      "35   taxon15  9    (1.112%)\t4\n",
      "36   taxon24  12   (1.483%)\t4\n",
      "37   taxon6   16   (1.978%)\t4\n",
      "**** TOTAL    624 (2.085%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 1091\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 1.824 MB RAM is required!\n",
      "Current best tree score: 1091\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 1084 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 1084 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 1084 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "RATCHET Iteration 30 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 40 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 50 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 1084 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  1 sec.\n",
      "NOTE: At the end, 14727 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "1084 1084 1084 1084 1084 \n",
      "BEST SCORE FOUND : 1084\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.splits.nex\n",
      "\n",
      "CPU time used for tree search: 2.455 sec (0h:0m:2s)\n",
      "Wall-clock time used for tree search: 2.463 sec (0h:0m:2s)\n",
      "Total CPU time used: 2.485 sec (0h:0m:2s)\n",
      "Total wall-clock time used: 2.493 sec (0h:0m:2s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.splits.nex...\n",
      "37 taxa and 351 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.contree\n",
      "Parsimony score of consensus tree: 1084\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/raw/10214_0/10214_0.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:35:34 2024\n",
      "Completed MPBOOT for: 10214_0\n",
      "##########################\n",
      "Running MPBOOT for: 27689_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/27689_0/27689_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/27689_0/27689_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/27689_0/27689_0\n",
      "Seed:    587751 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:34 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/27689_0/27689_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "WARNING: 16 sites contain only gaps or ambiguous chars.\n",
      "Alignment has 243 sequences with 2983 columns and 2031 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon261  1404 (47.067%)\t4\n",
      "2    taxon253  1409 (47.234%)\t4\n",
      "3    taxon86   880  (29.501%)\t4\n",
      "4    taxon34   1382 (46.329%)\t4\n",
      "5    taxon273  855  (28.662%)\t4\n",
      "6    taxon165  899  (30.137%)\t4\n",
      "7    taxon57   841  (28.193%)\t4\n",
      "8    taxon271  853  (28.595%)\t4\n",
      "9    taxon241  867  (29.065%)\t4\n",
      "10   taxon249  861  (28.864%)\t4\n",
      "11   taxon61   834  (27.958%)\t4\n",
      "12   taxon279  1408 (47.201%)\t4\n",
      "13   taxon79   890  (29.836%)\t4\n",
      "14   taxon132  882  (29.568%)\t4\n",
      "15   taxon40   771  (25.846%)\t4\n",
      "16   taxon209  849  (28.461%)\t4\n",
      "17   taxon210  1404 (47.067%)\t4\n",
      "18   taxon106  2684 (89.977%) !!!\t4\n",
      "19   taxon43   801  (26.852%)\t4\n",
      "20   taxon155  887  (29.735%)\t4\n",
      "21   taxon136  884  (29.635%)\t4\n",
      "22   taxon160  1353 (45.357%)\t4\n",
      "23   taxon93   1364 (45.726%)\t4\n",
      "24   taxon101  833  (27.925%)\t4\n",
      "25   taxon92   899  (30.137%)\t4\n",
      "26   taxon127  2222 (74.489%) !!!\t4\n",
      "27   taxon23   876  (29.366%)\t4\n",
      "28   taxon109  901  (30.204%)\t4\n",
      "29   taxon29   1753 (58.766%) !!!\t4\n",
      "30   taxon97   1839 (61.649%) !!!\t4\n",
      "31   taxon232  979  (32.819%)\t4\n",
      "32   taxon288  870  (29.165%)\t4\n",
      "33   taxon289  868  (29.098%)\t4\n",
      "34   taxon236  863  (28.931%)\t4\n",
      "35   taxon237  870  (29.165%)\t4\n",
      "36   taxon231  933  (31.277%)\t4\n",
      "37   taxon163  1364 (45.726%)\t4\n",
      "38   taxon64   1385 (46.430%)\t4\n",
      "39   taxon244  1758 (58.934%) !!!\t4\n",
      "40   taxon290  870  (29.165%)\t4\n",
      "41   taxon76   825  (27.657%)\t4\n",
      "42   taxon75   1754 (58.800%) !!!\t4\n",
      "43   taxon217  864  (28.964%)\t4\n",
      "44   taxon117  1576 (52.833%) !!!\t4\n",
      "45   taxon28   844  (28.294%)\t4\n",
      "46   taxon95   2192 (73.483%) !!!\t4\n",
      "47   taxon103  859  (28.797%)\t4\n",
      "48   taxon243  871  (29.199%)\t4\n",
      "49   taxon246  866  (29.031%)\t4\n",
      "50   taxon152  894  (29.970%)\t4\n",
      "51   taxon151  1337 (44.821%)\t4\n",
      "52   taxon125  975  (32.685%)\t4\n",
      "53   taxon207  1765 (59.169%) !!!\t4\n",
      "54   taxon239  862  (28.897%)\t4\n",
      "55   taxon41   1337 (44.821%)\t4\n",
      "56   taxon66   1303 (43.681%)\t4\n",
      "57   taxon116  1743 (58.431%) !!!\t4\n",
      "58   taxon258  852  (28.562%)\t4\n",
      "59   taxon256  850  (28.495%)\t4\n",
      "60   taxon68   1298 (43.513%)\t4\n",
      "61   taxon262  1390 (46.597%)\t4\n",
      "62   taxon254  1308 (43.848%)\t4\n",
      "63   taxon48   1401 (46.966%)\t4\n",
      "64   taxon276  858  (28.763%)\t4\n",
      "65   taxon12   853  (28.595%)\t4\n",
      "66   taxon266  856  (28.696%)\t4\n",
      "67   taxon102  816  (27.355%)\t4\n",
      "68   taxon39   1263 (42.340%)\t4\n",
      "69   taxon45   812  (27.221%)\t4\n",
      "70   taxon89   898  (30.104%)\t4\n",
      "71   taxon126  1587 (53.201%) !!!\t4\n",
      "72   taxon135  2521 (84.512%) !!!\t4\n",
      "73   taxon67   1370 (45.927%)\t4\n",
      "74   taxon131  1845 (61.850%) !!!\t4\n",
      "75   taxon124  900  (30.171%)\t4\n",
      "76   taxon219  2183 (73.181%) !!!\t4\n",
      "77   taxon6    836  (28.025%)\t4\n",
      "78   taxon226  936  (31.378%)\t4\n",
      "79   taxon227  855  (28.662%)\t4\n",
      "80   taxon230  858  (28.763%)\t4\n",
      "81   taxon211  1295 (43.413%)\t4\n",
      "82   taxon213  1300 (43.580%)\t4\n",
      "83   taxon218  1293 (43.346%)\t4\n",
      "84   taxon229  857  (28.729%)\t4\n",
      "85   taxon245  1297 (43.480%)\t4\n",
      "86   taxon157  2525 (84.646%) !!!\t4\n",
      "87   taxon130  904  (30.305%)\t4\n",
      "88   taxon139  1813 (60.778%) !!!\t4\n",
      "89   taxon269  1309 (43.882%)\t4\n",
      "90   taxon214  2190 (73.416%) !!!\t4\n",
      "91   taxon85   1352 (45.323%)\t4\n",
      "92   taxon49   833  (27.925%)\t4\n",
      "93   taxon166  860  (28.830%)\t4\n",
      "94   taxon167  860  (28.830%)\t4\n",
      "95   taxon83   1354 (45.391%)\t4\n",
      "96   taxon60   1744 (58.465%) !!!\t4\n",
      "97   taxon37   2175 (72.913%) !!!\t4\n",
      "98   taxon1    1210 (40.563%)\t4\n",
      "99   taxon248  941  (31.545%)\t4\n",
      "100  taxon252  914  (30.640%)\t4\n",
      "101  taxon119  878  (29.433%)\t4\n",
      "102  taxon278  886  (29.702%)\t4\n",
      "103  taxon159  905  (30.339%)\t4\n",
      "104  taxon111  912  (30.573%)\t4\n",
      "105  taxon122  921  (30.875%)\t4\n",
      "106  taxon31   841  (28.193%)\t4\n",
      "107  taxon120  2110 (70.734%) !!!\t4\n",
      "108  taxon283  891  (29.869%)\t4\n",
      "109  taxon282  888  (29.769%)\t4\n",
      "110  taxon62   855  (28.662%)\t4\n",
      "111  taxon80   892  (29.903%)\t4\n",
      "112  taxon11   2194 (73.550%) !!!\t4\n",
      "113  taxon7    2198 (73.684%) !!!\t4\n",
      "114  taxon142  1786 (59.873%) !!!\t4\n",
      "115  taxon158  1343 (45.022%)\t4\n",
      "116  taxon228  1767 (59.236%) !!!\t4\n",
      "117  taxon16   1745 (58.498%) !!!\t4\n",
      "118  taxon54   824  (27.623%)\t4\n",
      "119  taxon225  1453 (48.709%)\t4\n",
      "120  taxon44   863  (28.931%)\t4\n",
      "121  taxon112  1794 (60.141%) !!!\t4\n",
      "122  taxon212  1807 (60.577%) !!!\t4\n",
      "123  taxon73   900  (30.171%)\t4\n",
      "124  taxon242  2556 (85.686%) !!!\t4\n",
      "125  taxon8    1759 (58.967%) !!!\t4\n",
      "126  taxon51   982  (32.920%)\t4\n",
      "127  taxon115  2113 (70.835%) !!!\t4\n",
      "128  taxon24   812  (27.221%)\t4\n",
      "129  taxon25   1757 (58.900%) !!!\t4\n",
      "130  taxon17   1759 (58.967%) !!!\t4\n",
      "131  taxon216  1321 (44.284%)\t4\n",
      "132  taxon169  861  (28.864%)\t4\n",
      "133  taxon171  1371 (45.960%)\t4\n",
      "134  taxon138  884  (29.635%)\t4\n",
      "135  taxon250  859  (28.797%)\t4\n",
      "136  taxon42   766  (25.679%)\t4\n",
      "137  taxon53   1389 (46.564%)\t4\n",
      "138  taxon215  908  (30.439%)\t4\n",
      "139  taxon96   1787 (59.906%) !!!\t4\n",
      "140  taxon36   811  (27.187%)\t4\n",
      "141  taxon87   2117 (70.969%) !!!\t4\n",
      "142  taxon5    1368 (45.860%)\t4\n",
      "143  taxon84   1344 (45.055%)\t4\n",
      "144  taxon71   838  (28.093%)\t4\n",
      "145  taxon234  861  (28.864%)\t4\n",
      "146  taxon141  2172 (72.813%) !!!\t4\n",
      "147  taxon74   873  (29.266%)\t4\n",
      "148  taxon114  2114 (70.868%) !!!\t4\n",
      "149  taxon143  1787 (59.906%) !!!\t4\n",
      "150  taxon72   2095 (70.231%) !!!\t4\n",
      "151  taxon19   1713 (57.425%) !!!\t4\n",
      "152  taxon77   892  (29.903%)\t4\n",
      "153  taxon140  884  (29.635%)\t4\n",
      "154  taxon153  1346 (45.122%)\t4\n",
      "155  taxon33   767  (25.712%)\t4\n",
      "156  taxon156  937  (31.411%)\t4\n",
      "157  taxon137  896  (30.037%)\t4\n",
      "158  taxon150  903  (30.272%)\t4\n",
      "159  taxon94   1436 (48.139%)\t4\n",
      "160  taxon134  1881 (63.057%) !!!\t4\n",
      "161  taxon110  2109 (70.701%) !!!\t4\n",
      "162  taxon121  1426 (47.804%)\t4\n",
      "163  taxon154  1394 (46.731%)\t4\n",
      "164  taxon247  855  (28.662%)\t4\n",
      "165  taxon30   864  (28.964%)\t4\n",
      "166  taxon3    849  (28.461%)\t4\n",
      "167  taxon52   864  (28.964%)\t4\n",
      "168  taxon129  1862 (62.420%) !!!\t4\n",
      "169  taxon128  1608 (53.905%) !!!\t4\n",
      "170  taxon20   2132 (71.472%) !!!\t4\n",
      "171  taxon224  1763 (59.102%) !!!\t4\n",
      "172  taxon91   1358 (45.525%)\t4\n",
      "173  taxon118  933  (31.277%)\t4\n",
      "174  taxon26   1754 (58.800%) !!!\t4\n",
      "175  taxon88   1787 (59.906%) !!!\t4\n",
      "176  taxon47   818  (27.422%)\t4\n",
      "177  taxon148  903  (30.272%)\t4\n",
      "178  taxon2    850  (28.495%)\t4\n",
      "179  taxon284  868  (29.098%)\t4\n",
      "180  taxon35   1387 (46.497%)\t4\n",
      "181  taxon55   826  (27.690%)\t4\n",
      "182  taxon205  860  (28.830%)\t4\n",
      "183  taxon145  1343 (45.022%)\t4\n",
      "184  taxon32   1764 (59.135%) !!!\t4\n",
      "185  taxon104  1314 (44.050%)\t4\n",
      "186  taxon69   1827 (61.247%) !!!\t4\n",
      "187  taxon10   835  (27.992%)\t4\n",
      "188  taxon183  748  (25.075%)\t4\n",
      "189  taxon189  750  (25.142%)\t4\n",
      "190  taxon184  748  (25.075%)\t4\n",
      "191  taxon179  750  (25.142%)\t4\n",
      "192  taxon190  749  (25.109%)\t4\n",
      "193  taxon193  751  (25.176%)\t4\n",
      "194  taxon192  751  (25.176%)\t4\n",
      "195  taxon188  776  (26.014%)\t4\n",
      "196  taxon181  774  (25.947%)\t4\n",
      "197  taxon182  750  (25.142%)\t4\n",
      "198  taxon187  751  (25.176%)\t4\n",
      "199  taxon178  750  (25.142%)\t4\n",
      "200  taxon173  749  (25.109%)\t4\n",
      "201  taxon220  1190 (39.893%)\t4\n",
      "202  taxon222  1189 (39.859%)\t4\n",
      "203  taxon223  1189 (39.859%)\t4\n",
      "204  taxon201  751  (25.176%)\t4\n",
      "205  taxon180  815  (27.321%)\t4\n",
      "206  taxon280  888  (29.769%)\t4\n",
      "207  taxon281  887  (29.735%)\t4\n",
      "208  taxon98   2141 (71.773%) !!!\t4\n",
      "209  taxon82   1337 (44.821%)\t4\n",
      "210  taxon123  885  (29.668%)\t4\n",
      "211  taxon255  1379 (46.229%)\t4\n",
      "212  taxon90   1787 (59.906%) !!!\t4\n",
      "213  taxon9    1759 (58.967%) !!!\t4\n",
      "214  taxon38   805  (26.986%)\t4\n",
      "215  taxon149  1785 (59.839%) !!!\t4\n",
      "216  taxon58   1351 (45.290%)\t4\n",
      "217  taxon22   899  (30.137%)\t4\n",
      "218  taxon161  944  (31.646%)\t4\n",
      "219  taxon4    845  (28.327%)\t4\n",
      "220  taxon78   2207 (73.986%) !!!\t4\n",
      "221  taxon277  1408 (47.201%)\t4\n",
      "222  taxon21   2198 (73.684%) !!!\t4\n",
      "223  taxon263  854  (28.629%)\t4\n",
      "224  taxon100  1460 (48.944%)\t4\n",
      "225  taxon99   1337 (44.821%)\t4\n",
      "226  taxon59   1754 (58.800%) !!!\t4\n",
      "227  taxon18   831  (27.858%)\t4\n",
      "228  taxon268  863  (28.931%)\t4\n",
      "229  taxon164  1320 (44.251%)\t4\n",
      "230  taxon14   823  (27.590%)\t4\n",
      "231  taxon13   823  (27.590%)\t4\n",
      "232  taxon15   1359 (45.558%)\t4\n",
      "233  taxon240  903  (30.272%)\t4\n",
      "234  taxon113  2116 (70.935%) !!!\t4\n",
      "235  taxon168  1761 (59.035%) !!!\t4\n",
      "236  taxon133  1319 (44.217%)\t4\n",
      "237  taxon286  859  (28.797%)\t4\n",
      "238  taxon251  2124 (71.203%) !!!\t4\n",
      "239  taxon162  1309 (43.882%)\t4\n",
      "240  taxon259  853  (28.595%)\t4\n",
      "241  taxon260  878  (29.433%)\t4\n",
      "242  taxon70   886  (29.702%)\t4\n",
      "243  taxon50   818  (27.422%)\t4\n",
      "WARNING: 59 sequences contain more than 50% gaps/ambiguity\n",
      "**** TOTAL     298995 (41.248%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.001 seconds\n",
      "Parsimony score: 12871\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.002 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/27689_0/27689_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 86.535 MB RAM is required!\n",
      "Current best tree score: 12871\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "NOTE: REPS computation is segmented into 4 parts for speed...\n",
      "segment#1 ends at 192\n",
      "segment#2 ends at 528\n",
      "segment#3 ends at 1408\n",
      "segment#4 ends at 1721\n",
      "0 seconds.\n",
      "Completed MPBOOT for: 27689_0\n",
      "##########################\n",
      "Running MPBOOT for: 19782_1\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/19782_1/19782_1_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/19782_1/19782_1.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/19782_1/19782_1\n",
      "Seed:    959767 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:39 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/19782_1/19782_1_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "WARNING: 3 sites contain only gaps or ambiguous chars.\n",
      "Alignment has 25 sequences with 792 columns and 181 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon24  22   (2.778%)\t4\n",
      "2    taxon25  21   (2.652%)\t4\n",
      "3    taxon26  22   (2.778%)\t4\n",
      "4    taxon19  13   (1.641%)\t4\n",
      "5    taxon22  13   (1.641%)\t4\n",
      "6    taxon1   10   (1.263%)\t4\n",
      "7    taxon4   10   (1.263%)\t4\n",
      "8    taxon3   59   (7.449%)\t4\n",
      "9    taxon2   10   (1.263%)\t4\n",
      "10   taxon10  16   (2.020%)\t4\n",
      "11   taxon14  15   (1.894%)\t4\n",
      "12   taxon15  19   (2.399%)\t4\n",
      "13   taxon9   12   (1.515%)\t4\n",
      "14   taxon12  15   (1.894%)\t4\n",
      "15   taxon17  10   (1.263%)\t4\n",
      "16   taxon8   10   (1.263%)\t4\n",
      "17   taxon6   10   (1.263%)\t4\n",
      "18   taxon5   11   (1.389%)\t4\n",
      "19   taxon7   10   (1.263%)\t4\n",
      "20   taxon11  15   (1.894%)\t4\n",
      "21   taxon27  20   (2.525%)\t4\n",
      "22   taxon30  21   (2.652%)\t4\n",
      "23   taxon28  22   (2.778%)\t4\n",
      "24   taxon31  21   (2.652%)\t4\n",
      "25   taxon23  22   (2.778%)\t4\n",
      "**** TOTAL    429 (2.167%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 372\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/19782_1/19782_1.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.773 MB RAM is required!\n",
      "Current best tree score: 372\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 370 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 370 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 30 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 40 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 50 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 370 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "Completed MPBOOT for: 19782_1\n",
      "##########################\n",
      "Running MPBOOT for: 2225_0\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/2225_0/2225_0_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/2225_0/2225_0.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/2225_0/2225_0\n",
      "Seed:    620574 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:40 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/2225_0/2225_0_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 41 sequences with 1771 columns and 114 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon26  13   (0.734%)\t4\n",
      "2    taxon18  24   (1.355%)\t4\n",
      "3    taxon13  13   (0.734%)\t4\n",
      "4    taxon19  13   (0.734%)\t4\n",
      "5    taxon30  13   (0.734%)\t4\n",
      "6    taxon14  25   (1.412%)\t4\n",
      "7    taxon21  14   (0.791%)\t4\n",
      "8    taxon11  13   (0.734%)\t4\n",
      "9    taxon36  13   (0.734%)\t4\n",
      "10   taxon12  44   (2.484%)\t4\n",
      "11   taxon31  13   (0.734%)\t4\n",
      "12   taxon33  31   (1.750%)\t4\n",
      "13   taxon43  18   (1.016%)\t4\n",
      "14   taxon68  19   (1.073%)\t4\n",
      "15   taxon66  41   (2.315%)\t4\n",
      "16   taxon71  18   (1.016%)\t4\n",
      "17   taxon62  19   (1.073%)\t4\n",
      "18   taxon45  18   (1.016%)\t4\n",
      "19   taxon61  18   (1.016%)\t4\n",
      "20   taxon60  27   (1.525%)\t4\n",
      "21   taxon58  18   (1.016%)\t4\n",
      "22   taxon70  18   (1.016%)\t4\n",
      "23   taxon73  32   (1.807%)\t4\n",
      "24   taxon44  20   (1.129%)\t4\n",
      "25   taxon56  18   (1.016%)\t4\n",
      "26   taxon52  19   (1.073%)\t4\n",
      "27   taxon57  40   (2.259%)\t4\n",
      "28   taxon6   11   (0.621%)\t4\n",
      "29   taxon7   72   (4.065%)\t4\n",
      "30   taxon75  73   (4.122%)\t4\n",
      "31   taxon49  18   (1.016%)\t4\n",
      "32   taxon48  18   (1.016%)\t4\n",
      "33   taxon78  12   (0.678%)\t4\n",
      "34   taxon50  22   (1.242%)\t4\n",
      "35   taxon3   13   (0.734%)\t4\n",
      "36   taxon1   13   (0.734%)\t4\n",
      "37   taxon77  11   (0.621%)\t4\n",
      "38   taxon4   22   (1.242%)\t4\n",
      "39   taxon74  178  (10.051%)\t4\n",
      "40   taxon76  26   (1.468%)\t4\n",
      "41   taxon81  28   (1.581%)\t4\n",
      "**** TOTAL    1089 (1.500%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 59\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/2225_0/2225_0.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.829 MB RAM is required!\n",
      "Current best tree score: 59\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 59 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 59 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 59 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "RATCHET Iteration 30 / Score: 59 / Time: 0h:0m:0s (0h:0m:1s left)\n",
      "Completed MPBOOT for: 2225_0\n",
      "##########################\n",
      "Running MPBOOT for: 14553_1\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/14553_1/14553_1_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/14553_1/14553_1.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/14553_1/14553_1\n",
      "Seed:    111536 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:41 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14553_1/14553_1_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 28 sequences with 773 columns and 112 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon242  3    (0.388%)\t4\n",
      "2    taxon111  1    (0.129%)\t4\n",
      "3    taxon74   45   (5.821%)\t4\n",
      "4    taxon71   0    (0.000%)\t4\n",
      "5    taxon77   0    (0.000%)\t4\n",
      "6    taxon72   0    (0.000%)\t4\n",
      "7    taxon76   0    (0.000%)\t4\n",
      "8    taxon73   0    (0.000%)\t4\n",
      "9    taxon17   4    (0.517%)\t4\n",
      "10   taxon101  26   (3.364%)\t4\n",
      "11   taxon167  1    (0.129%)\t4\n",
      "12   taxon177  3    (0.388%)\t4\n",
      "13   taxon236  3    (0.388%)\t4\n",
      "14   taxon210  3    (0.388%)\t4\n",
      "15   taxon194  3    (0.388%)\t4\n",
      "16   taxon175  5    (0.647%)\t4\n",
      "17   taxon7    4    (0.517%)\t4\n",
      "18   taxon10   4    (0.517%)\t4\n",
      "19   taxon23   4    (0.517%)\t4\n",
      "20   taxon51   0    (0.000%)\t4\n",
      "21   taxon57   0    (0.000%)\t4\n",
      "22   taxon172  25   (3.234%)\t4\n",
      "23   taxon140  1    (0.129%)\t4\n",
      "24   taxon102  1    (0.129%)\t4\n",
      "25   taxon173  5    (0.647%)\t4\n",
      "26   taxon176  5    (0.647%)\t4\n",
      "27   taxon1    3    (0.388%)\t4\n",
      "28   taxon83   0    (0.000%)\t4\n",
      "**** TOTAL     149 (0.688%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 180\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/14553_1/14553_1.newick ...\n",
      "NOTE: Changing to old kernel as input tree is multifurcating\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.594 MB RAM is required!\n",
      "ERROR: A taxon has no name. (line 1 column 1268)\n",
      "Completed MPBOOT for: 14553_1\n",
      "##########################\n",
      "Running MPBOOT for: 19589_1\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/raw/19589_1/19589_1_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/raw/19589_1/19589_1.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/raw/19589_1/19589_1\n",
      "Seed:    117839 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:35:41 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/raw/19589_1/19589_1_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 416 sequences with 734 columns and 626 patterns\n",
      "ID   Sequence  #Gap/Ambiguity\n",
      "1    taxon515  124  (16.894%)\t4\n",
      "2    taxon516  125  (17.030%)\t4\n",
      "3    taxon512  124  (16.894%)\t4\n",
      "4    taxon513  123  (16.757%)\t4\n",
      "5    taxon66   124  (16.894%)\t4\n",
      "6    taxon69   125  (17.030%)\t4\n",
      "7    taxon73   124  (16.894%)\t4\n",
      "8    taxon74   124  (16.894%)\t4\n",
      "9    taxon172  124  (16.894%)\t4\n",
      "10   taxon171  99   (13.488%)\t4\n",
      "11   taxon168  58   (7.902%)\t4\n",
      "12   taxon165  124  (16.894%)\t4\n",
      "13   taxon170  51   (6.948%)\t4\n",
      "14   taxon167  124  (16.894%)\t4\n",
      "15   taxon166  124  (16.894%)\t4\n",
      "16   taxon366  88   (11.989%)\t4\n",
      "17   taxon368  112  (15.259%)\t4\n",
      "18   taxon367  112  (15.259%)\t4\n",
      "19   taxon370  133  (18.120%)\t4\n",
      "20   taxon369  124  (16.894%)\t4\n",
      "21   taxon372  151  (20.572%)\t4\n",
      "22   taxon373  88   (11.989%)\t4\n",
      "23   taxon41   124  (16.894%)\t4\n",
      "24   taxon40   124  (16.894%)\t4\n",
      "25   taxon39   124  (16.894%)\t4\n",
      "26   taxon56   124  (16.894%)\t4\n",
      "27   taxon48   259  (35.286%)\t4\n",
      "28   taxon49   163  (22.207%)\t4\n",
      "29   taxon51   133  (18.120%)\t4\n",
      "30   taxon54   194  (26.431%)\t4\n",
      "31   taxon47   124  (16.894%)\t4\n",
      "32   taxon57   131  (17.847%)\t4\n",
      "33   taxon55   484  (65.940%) !!!\t4\n",
      "34   taxon50   131  (17.847%)\t4\n",
      "35   taxon52   124  (16.894%)\t4\n",
      "36   taxon53   82   (11.172%)\t5\n",
      "37   taxon46   82   (11.172%)\t7\n",
      "38   taxon45   77   (10.490%)\t5\n",
      "39   taxon44   247  (33.651%)\t5\n",
      "40   taxon43   124  (16.894%)\t4\n",
      "41   taxon65   125  (17.030%)\t4\n",
      "42   taxon31   124  (16.894%)\t4\n",
      "43   taxon37   125  (17.030%)\t4\n",
      "44   taxon27   124  (16.894%)\t4\n",
      "45   taxon35   124  (16.894%)\t4\n",
      "46   taxon26   124  (16.894%)\t4\n",
      "47   taxon64   346  (47.139%)\t4\n",
      "48   taxon63   217  (29.564%)\t4\n",
      "49   taxon58   80   (10.899%)\t4\n",
      "50   taxon59   134  (18.256%)\t4\n",
      "51   taxon61   272  (37.057%)\t4\n",
      "52   taxon62   124  (16.894%)\t4\n",
      "53   taxon60   124  (16.894%)\t4\n",
      "54   taxon1    124  (16.894%)\t4\n",
      "55   taxon3    124  (16.894%)\t4\n",
      "56   taxon2    124  (16.894%)\t4\n",
      "57   taxon4    125  (17.030%)\t4\n",
      "58   taxon5    92   (12.534%)\t4\n",
      "59   taxon298  56   (7.629%)\t4\n",
      "60   taxon542  124  (16.894%)\t4\n",
      "61   taxon521  160  (21.798%)\t4\n",
      "62   taxon518  125  (17.030%)\t4\n",
      "63   taxon526  125  (17.030%)\t4\n",
      "64   taxon527  125  (17.030%)\t4\n",
      "65   taxon520  125  (17.030%)\t4\n",
      "66   taxon517  125  (17.030%)\t4\n",
      "67   taxon24   124  (16.894%)\t4\n",
      "68   taxon22   131  (17.847%)\t4\n",
      "69   taxon19   132  (17.984%)\t4\n",
      "70   taxon21   153  (20.845%)\t4\n",
      "71   taxon25   83   (11.308%)\t5\n",
      "72   taxon20   124  (16.894%)\t4\n",
      "73   taxon203  62   (8.447%)\t4\n",
      "74   taxon202  62   (8.447%)\t4\n",
      "75   taxon390  125  (17.030%)\t4\n",
      "76   taxon387  124  (16.894%)\t4\n",
      "77   taxon395  125  (17.030%)\t4\n",
      "78   taxon384  124  (16.894%)\t4\n",
      "79   taxon383  124  (16.894%)\t4\n",
      "80   taxon385  375  (51.090%) !!!\t4\n",
      "81   taxon402  124  (16.894%)\t4\n",
      "82   taxon394  125  (17.030%)\t4\n",
      "83   taxon80   62   (8.447%)\t4\n",
      "84   taxon79   62   (8.447%)\t4\n",
      "85   taxon381  125  (17.030%)\t4\n",
      "86   taxon382  124  (16.894%)\t4\n",
      "87   taxon375  124  (16.894%)\t4\n",
      "88   taxon374  124  (16.894%)\t4\n",
      "89   taxon376  124  (16.894%)\t4\n",
      "90   taxon377  124  (16.894%)\t4\n",
      "91   taxon379  259  (35.286%)\t4\n",
      "92   taxon378  125  (17.030%)\t4\n",
      "93   taxon316  192  (26.158%)\t4\n",
      "94   taxon317  302  (41.144%)\t4\n",
      "95   taxon321  221  (30.109%)\t4\n",
      "96   taxon322  106  (14.441%)\t4\n",
      "97   taxon323  95   (12.943%)\t4\n",
      "98   taxon320  101  (13.760%)\t4\n",
      "99   taxon319  227  (30.926%)\t4\n",
      "100  taxon318  89   (12.125%)\t4\n",
      "101  taxon315  124  (16.894%)\t4\n",
      "102  taxon329  124  (16.894%)\t4\n",
      "103  taxon337  125  (17.030%)\t4\n",
      "104  taxon347  70   (9.537%)\t4\n",
      "105  taxon348  124  (16.894%)\t4\n",
      "106  taxon349  279  (38.011%)\t4\n",
      "107  taxon324  124  (16.894%)\t4\n",
      "108  taxon371  602  (82.016%) !!!\t4\n",
      "109  taxon556  124  (16.894%)\t4\n",
      "110  taxon553  125  (17.030%)\t4\n",
      "111  taxon554  127  (17.302%)\t5\n",
      "112  taxon555  124  (16.894%)\t4\n",
      "113  taxon559  124  (16.894%)\t4\n",
      "114  taxon580  559  (76.158%) !!!\t4\n",
      "115  taxon121  124  (16.894%)\t4\n",
      "116  taxon114  131  (17.847%)\t4\n",
      "117  taxon106  124  (16.894%)\t4\n",
      "118  taxon108  137  (18.665%)\t4\n",
      "119  taxon105  124  (16.894%)\t4\n",
      "120  taxon117  124  (16.894%)\t4\n",
      "121  taxon103  77   (10.490%)\t4\n",
      "122  taxon102  77   (10.490%)\t4\n",
      "123  taxon107  72   (9.809%)\t4\n",
      "124  taxon230  124  (16.894%)\t4\n",
      "125  taxon223  124  (16.894%)\t4\n",
      "126  taxon246  125  (17.030%)\t4\n",
      "127  taxon264  181  (24.659%)\t4\n",
      "128  taxon261  123  (16.757%)\t4\n",
      "129  taxon227  62   (8.447%)\t4\n",
      "130  taxon237  124  (16.894%)\t4\n",
      "131  taxon224  124  (16.894%)\t4\n",
      "132  taxon239  124  (16.894%)\t4\n",
      "133  taxon234  124  (16.894%)\t4\n",
      "134  taxon269  124  (16.894%)\t4\n",
      "135  taxon268  386  (52.589%) !!!\t4\n",
      "136  taxon267  320  (43.597%)\t4\n",
      "137  taxon226  86   (11.717%)\t4\n",
      "138  taxon305  124  (16.894%)\t4\n",
      "139  taxon306  97   (13.215%)\t4\n",
      "140  taxon578  131  (17.847%)\t4\n",
      "141  taxon510  149  (20.300%)\t4\n",
      "142  taxon499  161  (21.935%)\t4\n",
      "143  taxon505  161  (21.935%)\t4\n",
      "144  taxon507  130  (17.711%)\t4\n",
      "145  taxon496  130  (17.711%)\t4\n",
      "146  taxon506  144  (19.619%)\t4\n",
      "147  taxon498  130  (17.711%)\t4\n",
      "148  taxon502  130  (17.711%)\t4\n",
      "149  taxon504  130  (17.711%)\t4\n",
      "150  taxon497  211  (28.747%)\t4\n",
      "151  taxon503  130  (17.711%)\t4\n",
      "152  taxon509  146  (19.891%)\t4\n",
      "153  taxon508  161  (21.935%)\t4\n",
      "154  taxon483  130  (17.711%)\t4\n",
      "155  taxon486  161  (21.935%)\t4\n",
      "156  taxon488  489  (66.621%) !!!\t4\n",
      "157  taxon484  304  (41.417%)\t4\n",
      "158  taxon485  130  (17.711%)\t4\n",
      "159  taxon487  130  (17.711%)\t4\n",
      "160  taxon133  115  (15.668%)\t4\n",
      "161  taxon134  88   (11.989%)\t4\n",
      "162  taxon132  80   (10.899%)\t4\n",
      "163  taxon136  80   (10.899%)\t4\n",
      "164  taxon135  62   (8.447%)\t4\n",
      "165  taxon217  97   (13.215%)\t4\n",
      "166  taxon218  105  (14.305%)\t4\n",
      "167  taxon219  95   (12.943%)\t4\n",
      "168  taxon216  54   (7.357%)\t4\n",
      "169  taxon215  74   (10.082%)\t4\n",
      "170  taxon214  69   (9.401%)\t4\n",
      "171  taxon210  110  (14.986%)\t4\n",
      "172  taxon209  112  (15.259%)\t4\n",
      "173  taxon211  97   (13.215%)\t4\n",
      "174  taxon212  95   (12.943%)\t4\n",
      "175  taxon494  353  (48.093%)\t4\n",
      "176  taxon495  352  (47.956%)\t4\n",
      "177  taxon424  124  (16.894%)\t4\n",
      "178  taxon416  87   (11.853%)\t4\n",
      "179  taxon417  187  (25.477%)\t4\n",
      "180  taxon418  110  (14.986%)\t4\n",
      "181  taxon414  129  (17.575%)\t4\n",
      "182  taxon411  225  (30.654%)\t4\n",
      "183  taxon405  225  (30.654%)\t4\n",
      "184  taxon412  226  (30.790%)\t4\n",
      "185  taxon407  223  (30.381%)\t4\n",
      "186  taxon409  241  (32.834%)\t4\n",
      "187  taxon410  293  (39.918%)\t4\n",
      "188  taxon408  230  (31.335%)\t4\n",
      "189  taxon406  222  (30.245%)\t4\n",
      "190  taxon423  101  (13.760%)\t4\n",
      "191  taxon420  85   (11.580%)\t4\n",
      "192  taxon415  80   (10.899%)\t4\n",
      "193  taxon422  80   (10.899%)\t4\n",
      "194  taxon421  80   (10.899%)\t4\n",
      "195  taxon425  100  (13.624%)\t4\n",
      "196  taxon437  130  (17.711%)\t4\n",
      "197  taxon429  130  (17.711%)\t4\n",
      "198  taxon433  130  (17.711%)\t4\n",
      "199  taxon430  130  (17.711%)\t4\n",
      "200  taxon435  130  (17.711%)\t4\n",
      "201  taxon428  130  (17.711%)\t4\n",
      "202  taxon431  130  (17.711%)\t4\n",
      "203  taxon436  135  (18.392%)\t4\n",
      "204  taxon426  130  (17.711%)\t4\n",
      "205  taxon434  130  (17.711%)\t4\n",
      "206  taxon427  130  (17.711%)\t4\n",
      "207  taxon438  276  (37.602%)\t4\n",
      "208  taxon307  124  (16.894%)\t4\n",
      "209  taxon481  85   (11.580%)\t4\n",
      "210  taxon480  78   (10.627%)\t5\n",
      "211  taxon479  80   (10.899%)\t4\n",
      "212  taxon12   77   (10.490%)\t4\n",
      "213  taxon15   86   (11.717%)\t4\n",
      "214  taxon14   77   (10.490%)\t4\n",
      "215  taxon13   238  (32.425%)\t4\n",
      "216  taxon16   197  (26.839%)\t4\n",
      "217  taxon18   78   (10.627%)\t4\n",
      "218  taxon17   80   (10.899%)\t4\n",
      "219  taxon7    125  (17.030%)\t4\n",
      "220  taxon6    137  (18.665%)\t4\n",
      "221  taxon176  70   (9.537%)\t4\n",
      "222  taxon177  157  (21.390%)\t4\n",
      "223  taxon174  51   (6.948%)\t4\n",
      "224  taxon178  51   (6.948%)\t4\n",
      "225  taxon179  51   (6.948%)\t4\n",
      "226  taxon81   80   (10.899%)\t4\n",
      "227  taxon82   78   (10.627%)\t4\n",
      "228  taxon154  91   (12.398%)\t7\n",
      "229  taxon155  214  (29.155%)\t4\n",
      "230  taxon152  302  (41.144%)\t4\n",
      "231  taxon151  322  (43.869%)\t8\n",
      "232  taxon141  62   (8.447%)\t4\n",
      "233  taxon148  124  (16.894%)\t4\n",
      "234  taxon142  63   (8.583%)\t4\n",
      "235  taxon145  94   (12.807%)\t4\n",
      "236  taxon144  105  (14.305%)\t4\n",
      "237  taxon139  102  (13.896%)\t4\n",
      "238  taxon140  62   (8.447%)\t4\n",
      "239  taxon149  62   (8.447%)\t4\n",
      "240  taxon153  62   (8.447%)\t4\n",
      "241  taxon147  62   (8.447%)\t4\n",
      "242  taxon150  62   (8.447%)\t4\n",
      "243  taxon143  124  (16.894%)\t4\n",
      "244  taxon163  152  (20.708%)\t4\n",
      "245  taxon162  110  (14.986%)\t4\n",
      "246  taxon160  89   (12.125%)\t4\n",
      "247  taxon159  92   (12.534%)\t4\n",
      "248  taxon158  124  (16.894%)\t4\n",
      "249  taxon164  124  (16.894%)\t4\n",
      "250  taxon84   124  (16.894%)\t4\n",
      "251  taxon89   124  (16.894%)\t4\n",
      "252  taxon87   124  (16.894%)\t4\n",
      "253  taxon88   130  (17.711%)\t4\n",
      "254  taxon91   124  (16.894%)\t4\n",
      "255  taxon92   124  (16.894%)\t4\n",
      "256  taxon543  125  (17.030%)\t4\n",
      "257  taxon99   124  (16.894%)\t4\n",
      "258  taxon96   124  (16.894%)\t4\n",
      "259  taxon97   81   (11.035%)\t4\n",
      "260  taxon98   87   (11.853%)\t4\n",
      "261  taxon95   88   (11.989%)\t5\n",
      "262  taxon94   58   (7.902%)\t4\n",
      "263  taxon93   62   (8.447%)\t4\n",
      "264  taxon573  125  (17.030%)\t4\n",
      "265  taxon572  125  (17.030%)\t4\n",
      "266  taxon570  125  (17.030%)\t4\n",
      "267  taxon571  125  (17.030%)\t4\n",
      "268  taxon574  129  (17.575%)\t4\n",
      "269  taxon576  124  (16.894%)\t4\n",
      "270  taxon575  124  (16.894%)\t4\n",
      "271  taxon545  62   (8.447%)\t4\n",
      "272  taxon550  157  (21.390%)\t4\n",
      "273  taxon548  58   (7.902%)\t4\n",
      "274  taxon549  57   (7.766%)\t4\n",
      "275  taxon547  62   (8.447%)\t4\n",
      "276  taxon544  62   (8.447%)\t4\n",
      "277  taxon546  81   (11.035%)\t4\n",
      "278  taxon560  80   (10.899%)\t4\n",
      "279  taxon561  60   (8.174%)\t4\n",
      "280  taxon564  55   (7.493%)\t4\n",
      "281  taxon563  77   (10.490%)\t6\n",
      "282  taxon562  81   (11.035%)\t5\n",
      "283  taxon359  55   (7.493%)\t4\n",
      "284  taxon360  99   (13.488%)\t4\n",
      "285  taxon364  230  (31.335%)\t4\n",
      "286  taxon362  233  (31.744%)\t4\n",
      "287  taxon361  247  (33.651%)\t4\n",
      "288  taxon365  260  (35.422%)\t6\n",
      "289  taxon363  235  (32.016%)\t4\n",
      "290  taxon358  75   (10.218%)\t4\n",
      "291  taxon355  234  (31.880%)\t4\n",
      "292  taxon356  119  (16.213%)\t4\n",
      "293  taxon357  230  (31.335%)\t4\n",
      "294  taxon353  53   (7.221%)\t4\n",
      "295  taxon352  53   (7.221%)\t4\n",
      "296  taxon350  233  (31.744%)\t4\n",
      "297  taxon354  287  (39.101%)\t4\n",
      "298  taxon351  286  (38.965%)\t4\n",
      "299  taxon194  62   (8.447%)\t4\n",
      "300  taxon193  62   (8.447%)\t4\n",
      "301  taxon199  62   (8.447%)\t4\n",
      "302  taxon201  201  (27.384%)\t4\n",
      "303  taxon198  80   (10.899%)\t4\n",
      "304  taxon196  67   (9.128%)\t4\n",
      "305  taxon195  85   (11.580%)\t4\n",
      "306  taxon197  56   (7.629%)\t4\n",
      "307  taxon200  62   (8.447%)\t4\n",
      "308  taxon551  124  (16.894%)\t4\n",
      "309  taxon184  62   (8.447%)\t4\n",
      "310  taxon186  88   (11.989%)\t4\n",
      "311  taxon181  105  (14.305%)\t4\n",
      "312  taxon182  116  (15.804%)\t4\n",
      "313  taxon188  202  (27.520%)\t4\n",
      "314  taxon191  196  (26.703%)\t4\n",
      "315  taxon192  204  (27.793%)\t4\n",
      "316  taxon190  209  (28.474%)\t4\n",
      "317  taxon187  80   (10.899%)\t4\n",
      "318  taxon189  239  (32.561%)\t4\n",
      "319  taxon183  72   (9.809%)\t4\n",
      "320  taxon204  125  (17.030%)\t4\n",
      "321  taxon205  124  (16.894%)\t4\n",
      "322  taxon206  130  (17.711%)\t4\n",
      "323  taxon207  124  (16.894%)\t4\n",
      "324  taxon208  74   (10.082%)\t4\n",
      "325  taxon11   78   (10.627%)\t4\n",
      "326  taxon10   100  (13.624%)\t4\n",
      "327  taxon9    257  (35.014%)\t4\n",
      "328  taxon446  84   (11.444%)\t4\n",
      "329  taxon444  103  (14.033%)\t4\n",
      "330  taxon442  96   (13.079%)\t4\n",
      "331  taxon443  107  (14.578%)\t4\n",
      "332  taxon441  145  (19.755%)\t4\n",
      "333  taxon445  138  (18.801%)\t4\n",
      "334  taxon440  153  (20.845%)\t4\n",
      "335  taxon439  196  (26.703%)\t4\n",
      "336  taxon274  140  (19.074%)\t4\n",
      "337  taxon275  154  (20.981%)\t4\n",
      "338  taxon270  99   (13.488%)\t4\n",
      "339  taxon273  66   (8.992%)\t4\n",
      "340  taxon272  97   (13.215%)\t4\n",
      "341  taxon271  104  (14.169%)\t4\n",
      "342  taxon285  88   (11.989%)\t4\n",
      "343  taxon293  202  (27.520%)\t4\n",
      "344  taxon282  243  (33.106%)\t5\n",
      "345  taxon284  247  (33.651%)\t4\n",
      "346  taxon292  223  (30.381%)\t4\n",
      "347  taxon283  228  (31.063%)\t5\n",
      "348  taxon281  80   (10.899%)\t4\n",
      "349  taxon291  69   (9.401%)\t4\n",
      "350  taxon290  57   (7.766%)\t4\n",
      "351  taxon289  356  (48.501%)\t4\n",
      "352  taxon287  86   (11.717%)\t4\n",
      "353  taxon279  77   (10.490%)\t4\n",
      "354  taxon280  81   (11.035%)\t4\n",
      "355  taxon301  187  (25.477%)\t6\n",
      "356  taxon300  77   (10.490%)\t4\n",
      "357  taxon303  80   (10.899%)\t4\n",
      "358  taxon302  58   (7.902%)\t4\n",
      "359  taxon299  85   (11.580%)\t4\n",
      "360  taxon491  83   (11.308%)\t7\n",
      "361  taxon482  284  (38.692%)\t4\n",
      "362  taxon403  101  (13.760%)\t4\n",
      "363  taxon404  438  (59.673%) !!!\t4\n",
      "364  taxon8    54   (7.357%)\t4\n",
      "365  taxon568  124  (16.894%)\t4\n",
      "366  taxon565  124  (16.894%)\t4\n",
      "367  taxon566  124  (16.894%)\t4\n",
      "368  taxon567  79   (10.763%)\t5\n",
      "369  taxon569  47   (6.403%)\t4\n",
      "370  taxon173  77   (10.490%)\t4\n",
      "371  taxon156  124  (16.894%)\t4\n",
      "372  taxon76   124  (16.894%)\t4\n",
      "373  taxon157  124  (16.894%)\t4\n",
      "374  taxon161  223  (30.381%)\t4\n",
      "375  taxon131  51   (6.948%)\t4\n",
      "376  taxon577  130  (17.711%)\t4\n",
      "377  taxon552  77   (10.490%)\t4\n",
      "378  taxon77   56   (7.629%)\t4\n",
      "379  taxon489  130  (17.711%)\t4\n",
      "380  taxon490  124  (16.894%)\t4\n",
      "381  taxon78   109  (14.850%)\t5\n",
      "382  taxon180  77   (10.490%)\t4\n",
      "383  taxon296  96   (13.079%)\t4\n",
      "384  taxon295  62   (8.447%)\t4\n",
      "385  taxon294  121  (16.485%)\t4\n",
      "386  taxon213  256  (34.877%)\t4\n",
      "387  taxon277  44   (5.995%)\t4\n",
      "388  taxon304  60   (8.174%)\t4\n",
      "389  taxon308  94   (12.807%)\t4\n",
      "390  taxon511  125  (17.030%)\t4\n",
      "391  taxon310  147  (20.027%)\t4\n",
      "392  taxon309  125  (17.030%)\t4\n",
      "393  taxon493  130  (17.711%)\t4\n",
      "394  taxon492  130  (17.711%)\t4\n",
      "395  taxon311  90   (12.262%)\t4\n",
      "396  taxon312  112  (15.259%)\t4\n",
      "397  taxon313  80   (10.899%)\t4\n",
      "398  taxon137  246  (33.515%)\t4\n",
      "399  taxon579  80   (10.899%)\t4\n",
      "400  taxon75   124  (16.894%)\t4\n",
      "401  taxon297  62   (8.447%)\t4\n",
      "402  taxon314  62   (8.447%)\t4\n",
      "403  taxon138  62   (8.447%)\t4\n",
      "404  taxon478  154  (20.981%)\t4\n",
      "405  taxon477  156  (21.253%)\t4\n",
      "406  taxon475  124  (16.894%)\t4\n",
      "407  taxon476  132  (17.984%)\t4\n",
      "408  taxon457  124  (16.894%)\t4\n",
      "409  taxon465  125  (17.030%)\t4\n",
      "410  taxon467  124  (16.894%)\t4\n",
      "411  taxon474  153  (20.845%)\t4\n",
      "412  taxon473  125  (17.030%)\t4\n",
      "413  taxon464  354  (48.229%)\t4\n",
      "414  taxon447  235  (32.016%)\t4\n",
      "415  taxon448  77   (10.490%)\t4\n",
      "416  taxon449  80   (10.899%)\t4\n",
      "WARNING: 7 sequences contain more than 50% gaps/ambiguity\n",
      "**** TOTAL     55566 (18.198%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.001 seconds\n",
      "Parsimony score: 3463\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.001 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/raw/19589_1/19589_1.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 46.044 MB RAM is required!\n",
      "Current best tree score: 3463\n",
      "\n",
      "Doing NNI on the initial tree ... \n"
     ]
    }
   ],
   "source": [
    "counter = 1\n",
    "for dir in os.listdir(file_dir):\n",
    "    print(\"##########################\")\n",
    "    # if(dir == \"362_1\"):\n",
    "    print(\"Running MPBOOT for:\", dir)\n",
    "    seq_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \"_msa.fasta\"\n",
    "    prefix = file_dir+ \"/\"+ dir+ \"/\" + dir \n",
    "    tree_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \".newick\"\n",
    "    boostrap_tree_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \".splits.nex\"\n",
    "    mpboot_cmd = f\"{mpboot_path} -s {seq_file} -bb 1000 {tree_file} -pre {prefix}\"\n",
    "    # iqtree_cmd = f\"iqtree2 -sup {tree_file} -t {boostrap_tree_file} --prefix {prefix}\"\n",
    "    # \n",
    "    os.system(mpboot_cmd)\n",
    "    # os.system(iqtree_cmd)\n",
    "    print(\"Completed MPBOOT for:\", dir)\n",
    "    \n",
    "    # print(seq_file)\n",
    "    # print(prefix)\n",
    "    # print(tree_file)\n",
    "    counter += 1\n",
    "print(counter)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2024-03-26T00:34:13.248461Z"
    }
   },
   "id": "359e4179e4d781a3",
   "execution_count": null
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running MPBOOT for: 688_1\n",
      "MPBoot version 1.1.1 for Linux 64-bit built Mar 20 2024\n",
      "Copyright (c) 2016 Diep Thi Hoang, Le Sy Vinh, Tomas Flouri, Alexandros Stamatakis, Arndt von Haeseler and Bui Quang Minh.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 16 GB RAM)\n",
      "Command: /home/piyumal/Tools/MPBOOT/compiled/mpboot -s /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1_msa.fasta -bb 1000 /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.newick -pre /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1\n",
      "Seed:    168212 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:46:19 2024\n",
      "Kernel:  AVX\n",
      "\n",
      "Reading alignment file /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1_msa.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 12 sequences with 756 columns and 107 patterns\n",
      "ID   Sequence #Gap/Ambiguity\n",
      "1    taxon1   37   (4.894%)\t4\n",
      "2    taxon7   13   (1.720%)\t4\n",
      "3    taxon8   14   (1.852%)\t4\n",
      "4    taxon6   13   (1.720%)\t4\n",
      "5    taxon9   14   (1.852%)\t4\n",
      "6    taxon11  14   (1.852%)\t4\n",
      "7    taxon12  13   (1.720%)\t4\n",
      "8    taxon10  17   (2.249%)\t4\n",
      "9    taxon5   18   (2.381%)\t4\n",
      "10   taxon2   14   (1.852%)\t4\n",
      "11   taxon4   17   (2.249%)\t4\n",
      "12   taxon3   19   (2.513%)\t4\n",
      "**** TOTAL    203 (2.238%)\n",
      "\n",
      "Reading user tree for evaluating patterns\n",
      "Time for reading: 0.000 seconds\n",
      "Parsimony score: 187\n",
      "Reordering patterns in alignment by decreasing order of pattern parsimony... 0.000 seconds\n",
      "\n",
      "Reading input tree file /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.newick ...\n",
      "Generating 1000 samples for bootstrap approximation...\n",
      "\n",
      "NOTE: 0.207 MB RAM is required!\n",
      "Current best tree score: 187\n",
      "\n",
      "Doing NNI on the initial tree ... \n",
      "Current best score: 185 / CPU time: 0\n",
      "\n",
      "Time: 0h:0m:0s\n",
      "RATCHET Iteration 10 / Score: 185 / Time: 0h:0m:0s\n",
      "RATCHET Iteration 20 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 30 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 40 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 50 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 60 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 70 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 80 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 90 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "RATCHET Iteration 100 / Score: 185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Optimizing bootstrap trees ...\n",
      "100 replicates done\n",
      "200 replicates done\n",
      "300 replicates done\n",
      "400 replicates done\n",
      "500 replicates done\n",
      "600 replicates done\n",
      "700 replicates done\n",
      "800 replicates done\n",
      "900 replicates done\n",
      "1000 replicates done\n",
      "CPU Time used:  0 sec.\n",
      "NOTE: At the end, 7096 bootstrap candidate trees evaluated.\n",
      "Scores of best 5 trees: \n",
      "185 185 185 185 185 \n",
      "BEST SCORE FOUND : 185\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.splits.nex\n",
      "\n",
      "CPU time used for tree search: 0.459 sec (0h:0m:0s)\n",
      "Wall-clock time used for tree search: 0.459 sec (0h:0m:0s)\n",
      "Total CPU time used: 0.463 sec (0h:0m:0s)\n",
      "Total wall-clock time used: 0.465 sec (0h:0m:0s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.splits.nex...\n",
      "12 taxa and 103 splits.\n",
      "Consensus tree written to /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.contree\n",
      "Parsimony score of consensus tree: 185\n",
      "\n",
      "Analysis results written to: \n",
      "  MPBoot report:                 /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.mpboot\n",
      "  Maximum-parsimony  tree:       /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.treefile\n",
      "\n",
      "Maximum parsimony bootstrap approximation results written to:\n",
      "  Split support values:          /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.splits.nex\n",
      "  Consensus tree:                /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.contree\n",
      "  Screen log file:               /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.log\n",
      "\n",
      "Date and Time: Tue Mar 26 11:46:19 2024\n",
      "Completed MPBOOT for: 688_1\n"
     ]
    }
   ],
   "source": [
    "file_dir = \"/home/piyumal/PHD/BootstrapSupportPrediction/sample\"\n",
    "dir = \"688_1\"\n",
    "print(\"Running MPBOOT for:\", dir)\n",
    "seq_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \"_msa.fasta\"\n",
    "prefix = file_dir+ \"/\"+ dir+ \"/\" + dir \n",
    "tree_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \".newick\"\n",
    "boostrap_tree_file = file_dir+ \"/\"+ dir+ \"/\" + dir + \".splits.nex\"\n",
    "mpboot_cmd = f\"{mpboot_path} -s {seq_file} -bb 1000 {tree_file} -pre {prefix}\"\n",
    "# iqtree_cmd = f\"iqtree2 -sup {tree_file} -t {boostrap_tree_file} --prefix {prefix}\"\n",
    "# \n",
    "os.system(mpboot_cmd)\n",
    "# os.system(iqtree_cmd)\n",
    "print(\"Completed MPBOOT for:\", dir)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T00:46:19.637198Z",
     "start_time": "2024-03-26T00:46:19.164945Z"
    }
   },
   "id": "6ca682fc5f5123d0",
   "execution_count": 8
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.6 for Linux 64-bit built May 30 2020\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    piyumal (AVX2, FMA3, 15 GB RAM)\n",
      "Command: iqtree2 -sup /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.newick -t /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.splits.nex --prefix /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1\n",
      "Seed:    249581 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Tue Mar 26 11:46:21 2024\n",
      "Kernel:  AVX+FMA - 1 threads (12 CPU cores detected)\n",
      "\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 12 cores!\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\n",
      "\n",
      "Reading tree /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.newick ...\n",
      "unrooted tree detected\n",
      "Reading input file /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.splits.nex...\n",
      "12 taxa and 103 splits.\n",
      "Rescaling split weights by 1.000\n",
      "103 splits found\n",
      "Creating bootstrap support values...\n",
      "Tree with assigned support written to /home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.suptree\n",
      "Date and Time: Tue Mar 26 11:46:21 2024\n"
     ]
    },
    {
     "data": {
      "text/plain": "0"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iqtree_cmd = f\"iqtree2 -sup {tree_file} -t {boostrap_tree_file} --prefix {prefix}\"\n",
    "os.system(iqtree_cmd)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T00:46:21.253511Z",
     "start_time": "2024-03-26T00:46:21.247865Z"
    }
   },
   "id": "85e432074552863a",
   "execution_count": 9
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "ename": "NewickError",
     "evalue": "Unexpected newick format '1/36:0.0013920000' \nYou may want to check other newick loading flags like 'format' or 'quoted_node_names'.",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mNewickError\u001B[0m                               Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[62], line 5\u001B[0m\n\u001B[1;32m      3\u001B[0m \u001B[38;5;28;01mwith\u001B[39;00m \u001B[38;5;28mopen\u001B[39m(iqtree_file, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mr\u001B[39m\u001B[38;5;124m\"\u001B[39m) \u001B[38;5;28;01mas\u001B[39;00m support_file:\n\u001B[1;32m      4\u001B[0m         tree_str \u001B[38;5;241m=\u001B[39m support_file\u001B[38;5;241m.\u001B[39mread()\n\u001B[0;32m----> 5\u001B[0m         tree \u001B[38;5;241m=\u001B[39m \u001B[43mTree\u001B[49m\u001B[43m(\u001B[49m\u001B[43mtree_str\u001B[49m\u001B[43m,\u001B[49m\u001B[43mquoted_node_names\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43;01mTrue\u001B[39;49;00m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;28;43mformat\u001B[39;49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;241;43m4\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[1;32m      6\u001B[0m         branch_id_counter \u001B[38;5;241m=\u001B[39m \u001B[38;5;241m0\u001B[39m\n\u001B[1;32m      7\u001B[0m         results \u001B[38;5;241m=\u001B[39m []\n",
      "File \u001B[0;32m~/PHD/Implementation/MLPhylogenetics/EBG-train/venv/lib/python3.10/site-packages/ete3/coretype/tree.py:212\u001B[0m, in \u001B[0;36mTreeNode.__init__\u001B[0;34m(self, newick, format, dist, support, name, quoted_node_names)\u001B[0m\n\u001B[1;32m    210\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m newick \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[1;32m    211\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_dist \u001B[38;5;241m=\u001B[39m \u001B[38;5;241m0.0\u001B[39m\n\u001B[0;32m--> 212\u001B[0m     \u001B[43mread_newick\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnewick\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mroot_node\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43m \u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;28;43mformat\u001B[39;49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mformat\u001B[39;49m\u001B[43m,\u001B[49m\n\u001B[1;32m    213\u001B[0m \u001B[43m                \u001B[49m\u001B[43mquoted_names\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mquoted_node_names\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[0;32m~/PHD/Implementation/MLPhylogenetics/EBG-train/venv/lib/python3.10/site-packages/ete3/parser/newick.py:266\u001B[0m, in \u001B[0;36mread_newick\u001B[0;34m(newick, root_node, format, quoted_names)\u001B[0m\n\u001B[1;32m    264\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m NewickError(\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mUnexisting tree file or Malformed newick tree structure.\u001B[39m\u001B[38;5;124m'\u001B[39m)\n\u001B[1;32m    265\u001B[0m     \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[0;32m--> 266\u001B[0m         \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[43m_read_newick_from_string\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnw\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mroot_node\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mmatcher\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;28;43mformat\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mquoted_names\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    268\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m    269\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m NewickError(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mnewick\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m argument must be either a filename or a newick string.\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n",
      "File \u001B[0;32m~/PHD/Implementation/MLPhylogenetics/EBG-train/venv/lib/python3.10/site-packages/ete3/parser/newick.py:341\u001B[0m, in \u001B[0;36m_read_newick_from_string\u001B[0;34m(nw, root_node, matcher, formatcode, quoted_names)\u001B[0m\n\u001B[1;32m    339\u001B[0m                 closing_internal \u001B[38;5;241m=\u001B[39m  closing_internal\u001B[38;5;241m.\u001B[39mrstrip(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124m;\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m    340\u001B[0m                 \u001B[38;5;66;03m# read internal node data and go up one level\u001B[39;00m\n\u001B[0;32m--> 341\u001B[0m                 \u001B[43m_read_node_data\u001B[49m\u001B[43m(\u001B[49m\u001B[43mclosing_internal\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mcurrent_parent\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43minternal\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mmatcher\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mformatcode\u001B[49m\u001B[43m)\u001B[49m\n\u001B[1;32m    342\u001B[0m                 current_parent \u001B[38;5;241m=\u001B[39m current_parent\u001B[38;5;241m.\u001B[39mup\n\u001B[1;32m    344\u001B[0m \u001B[38;5;66;03m# references in node names are replaced with quoted text before returning\u001B[39;00m\n",
      "File \u001B[0;32m~/PHD/Implementation/MLPhylogenetics/EBG-train/venv/lib/python3.10/site-packages/ete3/parser/newick.py:445\u001B[0m, in \u001B[0;36m_read_node_data\u001B[0;34m(subnw, current_node, node_type, matcher, formatcode)\u001B[0m\n\u001B[1;32m    443\u001B[0m         _parse_extra_features(node, data[\u001B[38;5;241m2\u001B[39m])\n\u001B[1;32m    444\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[0;32m--> 445\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m NewickError(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mUnexpected newick format \u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;132;01m%s\u001B[39;00m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m \u001B[39m\u001B[38;5;124m\"\u001B[39m \u001B[38;5;241m%\u001B[39msubnw[\u001B[38;5;241m0\u001B[39m:\u001B[38;5;241m50\u001B[39m])\n\u001B[1;32m    446\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m\n",
      "\u001B[0;31mNewickError\u001B[0m: Unexpected newick format '1/36:0.0013920000' \nYou may want to check other newick loading flags like 'format' or 'quoted_node_names'."
     ]
    }
   ],
   "source": [
    "iqtree_file = \"/home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.suptree\"\n",
    "raxml_support_file = \"/home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.raxml.support\"\n",
    "with open(iqtree_file, \"r\") as support_file:\n",
    "        tree_str = support_file.read()\n",
    "        tree = Tree(tree_str,quoted_node_names=True, format=4)\n",
    "        branch_id_counter = 0\n",
    "        results = []\n",
    "        \n",
    "        for node in tree.traverse():\n",
    "            print(node.support)\n"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T05:16:32.581937Z",
     "start_time": "2024-03-26T05:16:32.550803Z"
    }
   },
   "id": "57bb34d9f9c5a97f",
   "execution_count": 62
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(taxon1:0.1524920000,(taxon2:0.0064740000,((taxon3:0.0044300000,taxon4:0.0111080000)1/36:0.0013920000,taxon5:0.0072360000)1/59:0.0045400000)1/100:0.1324060000,(taxon6:0.0000010000,(((taxon7:0.0000010000,(taxon8:0.0013540000,taxon9:0.0000010000)1/22:0.0000010000)1/37:0.0000010000,taxon10:0.0250130000)1/14:0.0013460000,(taxon11:0.0000010000,taxon12:0.0000010000)1/37:0.0000010000)15:0.0013460000)1/100:0.0311800000)1;\n"
     ]
    }
   ],
   "source": [
    "tree_str1 = \"\"\n",
    "with open(iqtree_file, \"r\") as support_file:\n",
    "        tree_str1 = support_file.read()\n",
    "        print(tree_str1)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T05:16:43.540612Z",
     "start_time": "2024-03-26T05:16:43.537835Z"
    }
   },
   "id": "2efbaa0a1f8cc2d",
   "execution_count": 63
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "# index = 0\n",
    "# tree_data = []\n",
    "# \n",
    "# for i in tree_str1:\n",
    "#     index +=1\n",
    "#     local_bs = []\n",
    "#     tree_data.append(i)\n",
    "#     if i== \")\" and tree_str1[index]!=\";\":\n",
    "#         local_index = index\n",
    "#         while tree_str1[local_index]!= \":\":\n",
    "#             local_bs.append(tree_str1[local_index])\n",
    "#             local_index +=1\n",
    "#         print(local_bs)\n",
    "    "
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T05:16:45.819222Z",
     "start_time": "2024-03-26T05:16:45.816865Z"
    }
   },
   "id": "d848061666608097",
   "execution_count": 64
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1', '/', '3', '6']\n",
      "['1', '/', '5', '9']\n",
      "['1', '/', '1', '0', '0']\n",
      "['1', '/', '2', '2']\n",
      "['1', '/', '3', '7']\n",
      "['1', '/', '1', '4']\n",
      "['1', '/', '3', '7']\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "'/' is not in list",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mValueError\u001B[0m                                Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[70], line 12\u001B[0m\n\u001B[1;32m     10\u001B[0m     local_index \u001B[38;5;241m+\u001B[39m\u001B[38;5;241m=\u001B[39m\u001B[38;5;241m1\u001B[39m\n\u001B[1;32m     11\u001B[0m index \u001B[38;5;241m=\u001B[39m local_index\n\u001B[0;32m---> 12\u001B[0m local_index \u001B[38;5;241m=\u001B[39m \u001B[43mlocal_bs\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mindex\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43m/\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[1;32m     13\u001B[0m a \u001B[38;5;241m=\u001B[39m local_bs[:local_index]\n\u001B[1;32m     14\u001B[0m b \u001B[38;5;241m=\u001B[39m local_bs[local_index\u001B[38;5;241m+\u001B[39m\u001B[38;5;241m1\u001B[39m:]\n",
      "\u001B[0;31mValueError\u001B[0m: '/' is not in list"
     ]
    }
   ],
   "source": [
    "\n",
    "index = 0\n",
    "tree_data = []\n",
    "while index<len(tree_str1):\n",
    "    local_bs = []\n",
    "    tree_data.append(tree_str1[index])\n",
    "    if tree_str1[index]== \")\" and tree_str1[index+1]!=\" \":\n",
    "        local_index = index+1\n",
    "        while tree_str1[local_index]!= \":\":\n",
    "            local_bs.append(tree_str1[local_index])\n",
    "            local_index +=1\n",
    "        index = local_index\n",
    "        local_index = local_bs.index(\"/\")\n",
    "        a = local_bs[:local_index]\n",
    "        b = local_bs[local_index+1:]\n",
    "        n1 = \"\".join(a)\n",
    "        n2 = \"\".join(b)\n",
    "        n1 = int(n1)\n",
    "        n2 = int(n2)\n",
    "        mpboot_support = n1/n2\n",
    "        tree_data.append(str(mpboot_support))\n",
    "        tree_data.append(\":\")\n",
    "        print(local_bs)\n",
    "        \n",
    "    index +=1\n",
    "print(tree_data)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T05:27:32.928642Z",
     "start_time": "2024-03-26T05:27:32.914115Z"
    }
   },
   "id": "4479d4b75720f9be",
   "execution_count": 70
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1', '/', '3', '6']\n",
      "['1', '/', '5', '9']\n",
      "['1', '/', '1', '0', '0']\n",
      "['1', '/', '2', '2']\n",
      "['1', '/', '3', '7']\n",
      "['1', '/', '1', '4']\n",
      "['1', '/', '3', '7']\n",
      "['1', '5']\n",
      "['1', '/', '1', '0', '0']\n",
      "['1', ';']\n",
      "['(', 't', 'a', 'x', 'o', 'n', '1', ':', '0', '.', '1', '5', '2', '4', '9', '2', '0', '0', '0', '0', ',', '(', 't', 'a', 'x', 'o', 'n', '2', ':', '0', '.', '0', '0', '6', '4', '7', '4', '0', '0', '0', '0', ',', '(', '(', 't', 'a', 'x', 'o', 'n', '3', ':', '0', '.', '0', '0', '4', '4', '3', '0', '0', '0', '0', '0', ',', 't', 'a', 'x', 'o', 'n', '4', ':', '0', '.', '0', '1', '1', '1', '0', '8', '0', '0', '0', '0', ')', '0.027777777777777776', ':', '0', '.', '0', '0', '1', '3', '9', '2', '0', '0', '0', '0', ',', 't', 'a', 'x', 'o', 'n', '5', ':', '0', '.', '0', '0', '7', '2', '3', '6', '0', '0', '0', '0', ')', '0.01694915254237288', ':', '0', '.', '0', '0', '4', '5', '4', '0', '0', '0', '0', '0', ')', '0.01', ':', '0', '.', '1', '3', '2', '4', '0', '6', '0', '0', '0', '0', ',', '(', 't', 'a', 'x', 'o', 'n', '6', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ',', '(', '(', '(', 't', 'a', 'x', 'o', 'n', '7', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ',', '(', 't', 'a', 'x', 'o', 'n', '8', ':', '0', '.', '0', '0', '1', '3', '5', '4', '0', '0', '0', '0', ',', 't', 'a', 'x', 'o', 'n', '9', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ')', '0.045454545454545456', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ')', '0.02702702702702703', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ',', 't', 'a', 'x', 'o', 'n', '1', '0', ':', '0', '.', '0', '2', '5', '0', '1', '3', '0', '0', '0', '0', ')', '0.07142857142857142', ':', '0', '.', '0', '0', '1', '3', '4', '6', '0', '0', '0', '0', ',', '(', 't', 'a', 'x', 'o', 'n', '1', '1', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ',', 't', 'a', 'x', 'o', 'n', '1', '2', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ')', '0.02702702702702703', ':', '0', '.', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', ')', '1', '0', '.', '0', '0', '1', '3', '4', '6', '0', '0', '0', '0', ')', '0.01', ':', '0', '.', '0', '3', '1', '1', '8', '0', '0', '0', '0', '0', ')', ';']\n"
     ]
    }
   ],
   "source": [
    "index = 0\n",
    "tree_data = []\n",
    "while index<len(tree_str1):\n",
    "    local_bs = []\n",
    "    tree_data.append(tree_str1[index])\n",
    "    if tree_str1[index]== \")\" and tree_str1[index+1]!=\" \":\n",
    "        local_index = index+1\n",
    "        while local_index <= (len(tree_str1)-1) and tree_str1[local_index]!= \":\":\n",
    "            local_bs.append(tree_str1[local_index])\n",
    "            local_index +=1\n",
    "        index = local_index\n",
    "        if \"/\" in local_bs:\n",
    "            local_index = local_bs.index(\"/\")\n",
    "            a = local_bs[:local_index]\n",
    "            b = local_bs[local_index+1:]\n",
    "            n1 = \"\".join(a)\n",
    "            n2 = \"\".join(b)\n",
    "            n1 = int(n1)\n",
    "            n2 = int(n2)\n",
    "            mpboot_support = n1/n2\n",
    "            tree_data.append(str(mpboot_support))\n",
    "            tree_data.append(\":\")\n",
    "        else:\n",
    "             tree_data.append(str(1))\n",
    "        print(local_bs)\n",
    "        \n",
    "    index +=1\n",
    "# print(index)\n",
    "\n",
    "data_len = len(tree_data)\n",
    "\n",
    "# print(len(tree_data))\n",
    "while tree_data[data_len-1]!= \")\":\n",
    "    data_len -=1\n",
    "tree_data = tree_data[:data_len]\n",
    "tree_data.append(\";\")\n",
    "print(tree_data)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:29:35.940652Z",
     "start_time": "2024-03-26T08:29:35.934948Z"
    }
   },
   "id": "fa5eef1d10c72f82",
   "execution_count": 96
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "tree_revised = \"\".join(tree_data)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:29:47.206541Z",
     "start_time": "2024-03-26T08:29:47.204124Z"
    }
   },
   "id": "3604944a51d2cf7d",
   "execution_count": 97
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "# tree_revised = tree_revised[:-1]+\";\""
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:29:47.503110Z",
     "start_time": "2024-03-26T08:29:47.500132Z"
    }
   },
   "id": "3727dcffa80ca2b0",
   "execution_count": 98
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": "'(taxon1:0.1524920000,(taxon2:0.0064740000,((taxon3:0.0044300000,taxon4:0.0111080000)0.027777777777777776:0.0013920000,taxon5:0.0072360000)0.01694915254237288:0.0045400000)0.01:0.1324060000,(taxon6:0.0000010000,(((taxon7:0.0000010000,(taxon8:0.0013540000,taxon9:0.0000010000)0.045454545454545456:0.0000010000)0.02702702702702703:0.0000010000,taxon10:0.0250130000)0.07142857142857142:0.0013460000,(taxon11:0.0000010000,taxon12:0.0000010000)0.02702702702702703:0.0000010000)10.0013460000)0.01:0.0311800000);'"
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tree_revised"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:29:47.934615Z",
     "start_time": "2024-03-26T08:29:47.931427Z"
    }
   },
   "id": "5a2ee479c7371d06",
   "execution_count": 99
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 : 1.0\n",
      "3 : 0.01\n",
      "4 : 0.01\n",
      "6 : 0.01694915254237288\n",
      "8 : 10.001346\n",
      "9 : 0.027777777777777776\n",
      "11 : 0.07142857142857142\n",
      "12 : 0.02702702702702703\n",
      "15 : 0.02702702702702703\n",
      "20 : 0.045454545454545456\n"
     ]
    }
   ],
   "source": [
    "tree = Tree(tree_revised)\n",
    "\n",
    "branch_id = 0      \n",
    "for node in tree.traverse():\n",
    "    branch_id +=1\n",
    "    if node.support is not None and not node.is_leaf():\n",
    "        print(branch_id, \":\" , node.support)\n",
    "tree.write(outfile=\"/home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.suptree.ete3\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:35:08.823446Z",
     "start_time": "2024-03-26T08:35:08.819040Z"
    }
   },
   "id": "f6e33931208f71c",
   "execution_count": 104
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "tree_string_2 = \"((((taxon8:0.000001,(taxon17:0.000001,taxon28:0.000001)80:0.000001)73:0.007635,taxon32:0.000001)75:0.000001,taxon35:0.032085)68:0.016431,(taxon50:0.000001,taxon48:0.000001)87:0.033560,taxon38:0.000001);\""
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:30:47.686977Z",
     "start_time": "2024-03-26T08:30:47.684658Z"
    }
   },
   "id": "99d4c02846118262",
   "execution_count": 101
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "tree2 = Tree(tree_string_2)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:30:57.484125Z",
     "start_time": "2024-03-26T08:30:57.481553Z"
    }
   },
   "id": "620774dc4c03628b",
   "execution_count": 102
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.0\n",
      "68.0\n",
      "87.0\n",
      "75.0\n",
      "73.0\n",
      "80.0\n"
     ]
    }
   ],
   "source": [
    "for node in tree2.traverse():\n",
    "     if node.support is not None and not node.is_leaf():\n",
    "            print(node.support)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:30:57.825133Z",
     "start_time": "2024-03-26T08:30:57.821999Z"
    }
   },
   "id": "2107fc57ca8683b2",
   "execution_count": 103
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "def validate_support_values(file_path=\"\"):\n",
    "    iqtree_file = \"/home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.suptree.ete3\"\n",
    "    raxml_file = \"/home/piyumal/PHD/BootstrapSupportPrediction/sample/688_1/688_1.newick\"\n",
    "\n",
    "    tree_iqtree = Tree(iqtree_file)\n",
    "\n",
    "    iqtree_node_traversal = []\n",
    "    for node in tree_iqtree.traverse():\n",
    "        edge = node.get_edges()[0]\n",
    "        clade_1 = list(edge[0])\n",
    "        clade_2 = list(edge[1])\n",
    "        node_name = clade_1 + clade_2\n",
    "        node_name = [node.name for node in node_name]\n",
    "        node_name.sort()\n",
    "        node_name = ':'.join(node_name)\n",
    "        node.name = node_name\n",
    "        iqtree_node_traversal.append(node.name)\n",
    "\n",
    "    tree_raxml = Tree(raxml_file)\n",
    "\n",
    "    raxml_node_traversal = []\n",
    "    for node in tree_raxml.traverse():\n",
    "        edge = node.get_edges()[0]\n",
    "        clade_1 = list(edge[0])\n",
    "        clade_2 = list(edge[1])\n",
    "        node_name = clade_1 + clade_2\n",
    "        node_name = [node.name for node in node_name]\n",
    "        node_name.sort()\n",
    "        node_name = ':'.join(node_name)\n",
    "        node.name = node_name\n",
    "        raxml_node_traversal.append(node.name)\n",
    "\n",
    "    iqtree_diff = []\n",
    "    raxml_diff = []\n",
    "    for u in raxml_node_traversal:\n",
    "        if u not in iqtree_node_traversal:\n",
    "            iqtree_diff.append(u)\n",
    "\n",
    "    for v in iqtree_node_traversal:\n",
    "        if v not in raxml_node_traversal:\n",
    "            raxml_diff.append(v)\n",
    "    iqtree_diff.reverse()\n",
    "    raxml_node_traversal_v1 = copy.deepcopy(raxml_node_traversal)\n",
    "    iqtree_node_traversal_v1 = copy.deepcopy(iqtree_node_traversal)\n",
    "    for idx_iter, item in enumerate(iqtree_diff):\n",
    "        index_diff = raxml_node_traversal.index(item)\n",
    "        raxml_node_traversal[index_diff] = raxml_diff[idx_iter]\n",
    "\n",
    "    iqtree_idx = []\n",
    "    raxml_node_traversal_post = raxml_node_traversal\n",
    "    \n",
    "    raxml_idx = []\n",
    "    iqtree_node_traversal_post = iqtree_node_traversal[1:]\n",
    "\n",
    "    for val in iqtree_node_traversal[1:]:\n",
    "        iqtree_idx.append(raxml_node_traversal_post.index(val))\n",
    "    print(raxml_node_traversal)\n",
    "    print(iqtree_node_traversal)\n",
    "    print(iqtree_idx)\n",
    "    print(iqtree_diff)\n",
    "    print(raxml_diff)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:37:00.043401Z",
     "start_time": "2024-03-26T08:37:00.036917Z"
    }
   },
   "id": "1cc3f48834c08a92",
   "execution_count": 105
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['taxon1:taxon10:taxon11:taxon12:taxon2:taxon3:taxon4:taxon5:taxon6:taxon7:taxon8:taxon9', 'taxon10:taxon11:taxon12:taxon7:taxon8:taxon9', 'taxon10:taxon7:taxon8:taxon9', 'taxon11:taxon12', 'taxon10:taxon11:taxon12:taxon6:taxon7:taxon8:taxon9', 'taxon6', 'taxon7:taxon8:taxon9', 'taxon10', 'taxon11', 'taxon12', 'taxon1', 'taxon2:taxon3:taxon4:taxon5', 'taxon7', 'taxon8:taxon9', 'taxon2', 'taxon3:taxon4:taxon5', 'taxon8', 'taxon9', 'taxon3:taxon4', 'taxon5', 'taxon3', 'taxon4']\n",
      "['taxon1:taxon10:taxon11:taxon12:taxon2:taxon3:taxon4:taxon5:taxon6:taxon7:taxon8:taxon9', 'taxon1', 'taxon2:taxon3:taxon4:taxon5', 'taxon10:taxon11:taxon12:taxon6:taxon7:taxon8:taxon9', 'taxon2', 'taxon3:taxon4:taxon5', 'taxon6', 'taxon10:taxon11:taxon12:taxon7:taxon8:taxon9', 'taxon3:taxon4', 'taxon5', 'taxon10:taxon7:taxon8:taxon9', 'taxon11:taxon12', 'taxon3', 'taxon4', 'taxon7:taxon8:taxon9', 'taxon10', 'taxon11', 'taxon12', 'taxon7', 'taxon8:taxon9', 'taxon8', 'taxon9']\n",
      "[10, 11, 4, 14, 15, 5, 1, 18, 19, 2, 3, 20, 21, 6, 7, 8, 9, 12, 13, 16, 17]\n",
      "['taxon1:taxon2:taxon3:taxon4:taxon5', 'taxon1:taxon2:taxon3:taxon4:taxon5:taxon6']\n",
      "['taxon10:taxon11:taxon12:taxon6:taxon7:taxon8:taxon9', 'taxon10:taxon11:taxon12:taxon7:taxon8:taxon9']\n"
     ]
    }
   ],
   "source": [
    "validate_support_values()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-03-26T08:37:00.400844Z",
     "start_time": "2024-03-26T08:37:00.396895Z"
    }
   },
   "id": "681e8d352b37f775",
   "execution_count": 106
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false
   },
   "id": "6978c239f757258f"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
